{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`Last update at 2021-08-11`\n",
    "\n",
    "# Outline\n",
    "\n",
    "```\n",
    "    1) Downloading (meta)genomes and metagenome-assembled genomes (MAGs) that contain a paired untargeted \n",
    "    metabolomics (LC-MS/MS and in the future GC-MS/MS)(metabolomic files also going to be downloaded):\n",
    "    1.1) Downloading all JSON files from PoDP;\n",
    "    1.2) Parsing JSON files and creating podp_merged_df;\n",
    "    1.3) Obtaining uniform Genbank code from BioSample ID; (estimated runtime of 02:46:30)\n",
    "    1.4) Adding new column with the downloaded NCBI IDs;\n",
    "    1.5) Downloading and unzipping the NCBI genomes;\n",
    "    1.6) Copying previously downloaded genomes;\n",
    "    1.7) Downloading metagenomic reads;\n",
    "    1.8) Downloading JGI (meta)genomes;\n",
    "    1.9) Separating Genomes without antiSMASH;\n",
    "    1.10) Running antiSMASH;\n",
    "    1.11) Filtering links for (meta)genomes with antiSMASH and downloading LC-MS/MS;\n",
    "    1.12) Download missing Gerwick cyanoibacterial MS/MS and selecting Gerwick mgf files.\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import time\n",
    "import glob\n",
    "import numpy as np\n",
    "import csv\n",
    "import subprocess\n",
    "import glob\n",
    "import os\n",
    "import re\n",
    "import networkx\n",
    "from networkx.algorithms.components.connected import connected_components\n",
    "from collections import defaultdict\n",
    "import matplotlib.pyplot as plt\n",
    "from Bio import SeqIO\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "from sklearn.neighbors import NearestNeighbors\n",
    "import pickle\n",
    "import json\n",
    "from pyteomics import mgf, auxiliary\n",
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !mkdir ./inputs\n",
    "# !mkdir ./outputs/\n",
    "# !mkdir ./temp_files/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Downloading genomes from the Paired omics Data Platform (PoDP)\n",
    "\n",
    "## 1.1. Downloading all JSON files from PoDP\n",
    "\n",
    "```\n",
    "Go to PoDP (https://pairedomicsdata.bioinformatics.nl) and download the database JSON files, move files to \n",
    "input folder (path/to/NPOmix/inputs/databases/).\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.2. Parsing JSON files and creating podp_merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_strain_dicts(filename):\n",
    "    strain_gbk_dict = {}\n",
    "    strain_biosample_dict = {}\n",
    "    strain_db_dict = {}\n",
    "    strain_to_shotgun = {}\n",
    "    with open(filename) as f:\n",
    "        data1 = json.load(f)\n",
    "        if 'message' not in data1.keys():\n",
    "            data1 = data1['genomes']\n",
    "            json_df1 = pd.DataFrame(data1)\n",
    "            for i,r in json_df1.iterrows():\n",
    "                genome_line = dict(json_df1['genome_ID'].loc[i])\n",
    "                if 'ENA_NCBI_accession' in genome_line.keys():\n",
    "                    genomeID = genome_line['ENA_NCBI_accession']\n",
    "                    db_id = 'ENA_NCBI_accession'\n",
    "                if 'GenBank_accession' in genome_line.keys():\n",
    "                    genomeID = genome_line['GenBank_accession']\n",
    "                    db_id = 'GenBank_accession'\n",
    "                if 'RefSeq_accession' in genome_line.keys():\n",
    "                    genomeID = genome_line['RefSeq_accession']\n",
    "                    db_id = 'RefSeq_accession'\n",
    "                if 'JGI_Genome_ID' in genome_line.keys():\n",
    "                    genomeID = genome_line['JGI_Genome_ID']\n",
    "                    db_id = 'JGI_Genome_ID'\n",
    "                if 'JGI_ID' in genome_line.keys():\n",
    "                    genomeID = genome_line['JGI_ID']\n",
    "                    db_id = 'JGI_ID'\n",
    "                if 'JGI_IMG_genome_ID'in genome_line.keys():\n",
    "                    genomeID = genome_line['JGI_IMG_genome_ID']\n",
    "                    db_id = 'JGI_IMG_genome_ID'\n",
    "                strain_to_shotgun[json_df1['genome_label'].loc[i]] = genome_line['genome_type']\n",
    "                strain_gbk_dict[json_df1['genome_label'].loc[i]] = genomeID\n",
    "                strain_db_dict[json_df1['genome_label'].loc[i]] = db_id\n",
    "                if 'BioSample_accession' in json_df1.columns:\n",
    "                    strain_biosample_dict[json_df1['genome_label'].loc[i]] = json_df1['BioSample_accession'].loc[i]\n",
    "                else:\n",
    "                    strain_biosample_dict[json_df1['genome_label'].loc[i]] = 'No BioSample'\n",
    "    return strain_gbk_dict,strain_biosample_dict,strain_db_dict,strain_to_shotgun\n",
    "\n",
    "def get_paired_df(filename,strain_gbk_dict,strain_biosample_dict,strain_db_dict,strain_to_shotgun):\n",
    "    col1,col2,col3,col4,col5,col6 = [],[],[],[],[],[]\n",
    "    with open(filename) as f:\n",
    "        data2 = json.load(f)['genome_metabolome_links']\n",
    "        json_df2 = pd.DataFrame(data2)\n",
    "        for i,r in json_df2.iterrows():\n",
    "            col1.append(strain_gbk_dict[json_df2['genome_label'].loc[i]])\n",
    "            col2.append(strain_db_dict[json_df2['genome_label'].loc[i]])\n",
    "            col3.append(strain_biosample_dict[json_df2['genome_label'].loc[i]])\n",
    "            col4.append(json_df2['metabolomics_file'].loc[i])\n",
    "            col5.append(json_df2['genome_label'].loc[i])\n",
    "            col6.append(strain_to_shotgun[json_df2['genome_label'].loc[i]])\n",
    "    frames = {'NCBI_ID':col1, 'Database':col2, 'BioSample':col3, \n",
    "              'LCMS_file':col4, 'Genome_label':col5, 'Method':col6}\n",
    "    paired_df = pd.DataFrame(frames)\n",
    "    return paired_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>Database</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>LCMS_file</th>\n",
       "      <th>Genome_label</th>\n",
       "      <th>Method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4891</th>\n",
       "      <td>GCA_000286575.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN00255227</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia multivorans CF2</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4892</th>\n",
       "      <td>GCA_000959505.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN03144971</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia dolosa AU0158</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4893</th>\n",
       "      <td>GCA_000959505.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN03144971</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia dolosa AU0158</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4894</th>\n",
       "      <td>GCA_003568605.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN02866398</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia thailandensis E264</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4895</th>\n",
       "      <td>GCA_003568605.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN02866398</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia thailandensis E264</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4896 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              NCBI_ID           Database     BioSample  \\\n",
       "0     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "1     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "2     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "3     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "4     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "...               ...                ...           ...   \n",
       "4891  GCA_000286575.1  GenBank_accession  SAMN00255227   \n",
       "4892  GCA_000959505.1  GenBank_accession  SAMN03144971   \n",
       "4893  GCA_000959505.1  GenBank_accession  SAMN03144971   \n",
       "4894  GCA_003568605.1  GenBank_accession  SAMN02866398   \n",
       "4895  GCA_003568605.1  GenBank_accession  SAMN02866398   \n",
       "\n",
       "                                              LCMS_file  \\\n",
       "0     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "1     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "2     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "3     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "4     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "...                                                 ...   \n",
       "4891  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4892  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4893  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4894  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4895  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "\n",
       "                         Genome_label  Method  \n",
       "0                         S. cattleya  genome  \n",
       "1                         S. cattleya  genome  \n",
       "2                         S. cattleya  genome  \n",
       "3                         S. cattleya  genome  \n",
       "4                         S. cattleya  genome  \n",
       "...                               ...     ...  \n",
       "4891     Burkholderia multivorans CF2  genome  \n",
       "4892       Burkholderia dolosa AU0158  genome  \n",
       "4893       Burkholderia dolosa AU0158  genome  \n",
       "4894  Burkholderia thailandensis E264  genome  \n",
       "4895  Burkholderia thailandensis E264  genome  \n",
       "\n",
       "[4896 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_list = glob.glob('./inputs/PoDP_datasets/*.json')\n",
    "\n",
    "for filename in table_list:\n",
    "    strain_gbk_dict,strain_biosample_dict,strain_db_dict,strain_to_shotgun = get_strain_dicts(filename)\n",
    "    if strain_gbk_dict:\n",
    "        if filename == table_list[0]:\n",
    "            podp_merged_df = get_paired_df(filename,strain_gbk_dict,strain_biosample_dict,strain_db_dict,strain_to_shotgun)\n",
    "        else:\n",
    "            paired_df = get_paired_df(filename,strain_gbk_dict,strain_biosample_dict,strain_db_dict,strain_to_shotgun)\n",
    "            frames = [podp_merged_df,paired_df]\n",
    "            podp_merged_df = pd.concat(frames)\n",
    "        \n",
    "podp_merged_df = podp_merged_df.reset_index(drop=True)\n",
    "\n",
    "podp_merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(table_list) ### contain all online datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "online_massive = ['MTBLS1606','MSV000086050','MSV000085589','MSV000085586','MSV000085376','MSV000085214','MSV000085210','MSV000085192','MSV000085180','MSV000085179','MSV000085159','MSV000085158','MSV000085141','MSV000085123','MSV000085085','MSV000085038','MSV000085032','MSV000085031','MSV000085027','MSV000085026','MSV000085023','MSV000085021','MSV000085018','MSV000085003','MSV000085000','MSV000084989','MSV000084954','MSV000084950','MSV000084950','MSV000084945','MSV000084884','MSV000084781','MSV000084771','MSV000084723','MSV000084674','MSV000084475','MSV000084475','MSV000084117','MSV000083835','MSV000083734','MSV000083648','MSV000083387','MSV000083302','MSV000083295','MSV000083268','MSV000083081','MSV000082988','MSV000082969','MSV000082891','MSV000082831','MSV000082285','MSV000082045','MSV000081832','MSV000081504','MSV000081318','MSV000081063','MSV000080427','MSV000080251','MSV000080179','MSV000079519','MSV000079284','MSV000079139','MSV000079015','MSV000078995','MSV000078891','MSV000078850','MSV000078847','MSV000078839','MSV000078836','MSV000078667','MSV000078556']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "downloaded_massive = []\n",
    "\n",
    "for item in podp_merged_df['LCMS_file']:\n",
    "    if 'MSV' in item:\n",
    "        if 'ftp://maftp' in item:\n",
    "            downloaded_massive.append(item[31:43])\n",
    "        else:\n",
    "            downloaded_massive.append(item[23:35])\n",
    "    if 'MTBLS' in item:\n",
    "        downloaded_massive.append(item[35:])\n",
    "    \n",
    "downloaded_massive = list(np.unique(downloaded_massive))\n",
    "\n",
    "for item in online_massive:\n",
    "    if item not in downloaded_massive:\n",
    "        print(item)\n",
    "### we downloaded and processed all datasets (69 unique and 71 total MassIVE/Metabolights IDs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2289, 4896)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(podp_merged_df['NCBI_ID'])),len(podp_merged_df)\n",
    "### unique genomes and total number of links (the second matches the online record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "799"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(podp_merged_df[podp_merged_df['BioSample'] == 'No BioSample'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1677"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(podp_merged_df[podp_merged_df['BioSample'] != 'No BioSample']['BioSample']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2476"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "799+1677"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENA_NCBI_accession 1298\n",
      "GenBank_accession 2107\n",
      "JGI_Genome_ID 765\n",
      "JGI_ID 2\n",
      "JGI_IMG_genome_ID 612\n",
      "RefSeq_accession 112\n"
     ]
    }
   ],
   "source": [
    "for item in np.unique(podp_merged_df['Database']):\n",
    "    print(item, len(podp_merged_df[podp_merged_df['Database'] == item]['NCBI_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genome 3456\n",
      "metagenome 1306\n",
      "metagenome-assembled genome 134\n"
     ]
    }
   ],
   "source": [
    "for item in np.unique(podp_merged_df['Method']):\n",
    "    print(item, len(podp_merged_df[podp_merged_df['Method'] == item]['NCBI_ID']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.3. Obtaining uniform Genbank code from BioSample ID\n",
    "\n",
    "```\n",
    "Separating JGI/ENA IDs from NCBI genomes, these will be downloaded in another step, for now we'll focus on NCBI genomes\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>Database</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>LCMS_file</th>\n",
       "      <th>Genome_label</th>\n",
       "      <th>Method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2115</th>\n",
       "      <td>GCA_000286575.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN00255227</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia multivorans CF2</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2116</th>\n",
       "      <td>GCA_000959505.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN03144971</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia dolosa AU0158</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2117</th>\n",
       "      <td>GCA_000959505.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN03144971</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia dolosa AU0158</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2118</th>\n",
       "      <td>GCA_003568605.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN02866398</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia thailandensis E264</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2119</th>\n",
       "      <td>GCA_003568605.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN02866398</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia thailandensis E264</td>\n",
       "      <td>genome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2120 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              NCBI_ID           Database     BioSample  \\\n",
       "0     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "1     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "2     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "3     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "4     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "...               ...                ...           ...   \n",
       "2115  GCA_000286575.1  GenBank_accession  SAMN00255227   \n",
       "2116  GCA_000959505.1  GenBank_accession  SAMN03144971   \n",
       "2117  GCA_000959505.1  GenBank_accession  SAMN03144971   \n",
       "2118  GCA_003568605.1  GenBank_accession  SAMN02866398   \n",
       "2119  GCA_003568605.1  GenBank_accession  SAMN02866398   \n",
       "\n",
       "                                              LCMS_file  \\\n",
       "0     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "1     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "2     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "3     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "4     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "...                                                 ...   \n",
       "2115  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "2116  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "2117  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "2118  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "2119  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "\n",
       "                         Genome_label  Method  \n",
       "0                         S. cattleya  genome  \n",
       "1                         S. cattleya  genome  \n",
       "2                         S. cattleya  genome  \n",
       "3                         S. cattleya  genome  \n",
       "4                         S. cattleya  genome  \n",
       "...                               ...     ...  \n",
       "2115     Burkholderia multivorans CF2  genome  \n",
       "2116       Burkholderia dolosa AU0158  genome  \n",
       "2117       Burkholderia dolosa AU0158  genome  \n",
       "2118  Burkholderia thailandensis E264  genome  \n",
       "2119  Burkholderia thailandensis E264  genome  \n",
       "\n",
       "[2120 rows x 6 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_df = podp_merged_df[podp_merged_df['Database'] != 'JGI_IMG_genome_ID']\n",
    "filtered_df = filtered_df[filtered_df['Database'] != 'JGI_Genome_ID']\n",
    "filtered_df = filtered_df[filtered_df['Database'] != 'JGI_ID']\n",
    "filtered_df = filtered_df[filtered_df['Database'] != 'ENA_NCBI_accession']\n",
    "filtered_df = filtered_df[filtered_df['BioSample'] != 'No BioSample']\n",
    "filtered_df = filtered_df.reset_index(drop=True)\n",
    "\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "### only run once\n",
    "\n",
    "# with open('./temp_files/genome_list_temp.txt', \"w\") as output:\n",
    "#     writer = csv.writer(output, lineterminator='\\n')\n",
    "#     for val in list(filtered_df['NCBI_ID']):\n",
    "#         writer.writerow([val])\n",
    "        \n",
    "# with open('./temp_files/biosample_list_temp.txt', \"w\") as output:\n",
    "#     writer = csv.writer(output, lineterminator='\\n')\n",
    "#     for val in list(filtered_df['BioSample']):\n",
    "#         writer.writerow([val])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "### only run once\n",
    "\n",
    "# start = time.time()\n",
    "\n",
    "# subprocess.call('python NCBI_getGenBankID.py --input ./temp_files/biosample_list_temp.txt --output ./outputs/biosample_list-round3-210220.out.csv', shell=True)\n",
    "\n",
    "# end = time.time()\n",
    "# hours, rem = divmod(end-start, 3600)\n",
    "# minutes, seconds = divmod(rem, 60)\n",
    "# print(\"{:0>2}:{:0>2}:{:05.2f}\".format(int(hours),int(minutes),seconds))\n",
    "### 02:46:30.68"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession_id</th>\n",
       "      <th>Assembly_id</th>\n",
       "      <th>GenBank_id</th>\n",
       "      <th>Refseq_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>376848</td>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>GCF_000240165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMN14411170</td>\n",
       "      <td>6465981</td>\n",
       "      <td>GCA_011765705.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMN14411171</td>\n",
       "      <td>6466001</td>\n",
       "      <td>GCA_011765735.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMD00106498</td>\n",
       "      <td>2168011</td>\n",
       "      <td>GCA_003945305.1</td>\n",
       "      <td>GCF_003945305.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMN05710194</td>\n",
       "      <td>1533101</td>\n",
       "      <td>GCA_002897315.1</td>\n",
       "      <td>GCF_002897315.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>SAMN03144971</td>\n",
       "      <td>315111</td>\n",
       "      <td>GCA_000959505.1</td>\n",
       "      <td>GCF_000959505.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>SAMN00623032</td>\n",
       "      <td>39588</td>\n",
       "      <td>GCA_000018505.1</td>\n",
       "      <td>GCF_000018505.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>SAMN03140189</td>\n",
       "      <td>315121</td>\n",
       "      <td>GCA_000959525.1</td>\n",
       "      <td>GCF_000959525.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>SAMN00255227</td>\n",
       "      <td>492528</td>\n",
       "      <td>GCA_000286575.1</td>\n",
       "      <td>GCF_000286575.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>SAMN02866398</td>\n",
       "      <td>1978921</td>\n",
       "      <td>GCA_003568605.1</td>\n",
       "      <td>GCF_003568605.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>237 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Accession_id Assembly_id       GenBank_id        Refseq_id\n",
       "0    SAMN02603879      376848  GCA_000240165.1  GCF_000240165.1\n",
       "1    SAMN14411170     6465981  GCA_011765705.1              NaN\n",
       "2    SAMN14411171     6466001  GCA_011765735.1              NaN\n",
       "3    SAMD00106498     2168011  GCA_003945305.1  GCF_003945305.1\n",
       "4    SAMN05710194     1533101  GCA_002897315.1  GCF_002897315.1\n",
       "..            ...         ...              ...              ...\n",
       "232  SAMN03144971      315111  GCA_000959505.1  GCF_000959505.1\n",
       "233  SAMN00623032       39588  GCA_000018505.1  GCF_000018505.1\n",
       "234  SAMN03140189      315121  GCA_000959525.1  GCF_000959525.1\n",
       "235  SAMN00255227      492528  GCA_000286575.1  GCF_000286575.1\n",
       "236  SAMN02866398     1978921  GCA_003568605.1  GCF_003568605.1\n",
       "\n",
       "[237 rows x 4 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biosample_df = pd.read_csv('./outputs/biosample_list-round3-210220.out.csv',sep='\\t')\n",
    "biosample_df = biosample_df[biosample_df['GenBank_id'] != '-']\n",
    "biosample_df = biosample_df.reset_index(drop=True)\n",
    "\n",
    "\n",
    "biosample_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.4. Adding new column with the downloaded NCBI IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4896, 2120, 259, 237)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(podp_merged_df),len(filtered_df),len(np.unique(filtered_df['BioSample'])),len(np.unique(biosample_df['GenBank_id']))\n",
    "### notice that we downloaded 237 out of the 259 unique BioSamples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>Database</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>LCMS_file</th>\n",
       "      <th>Genome_label</th>\n",
       "      <th>Method</th>\n",
       "      <th>New_NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000240165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000240165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000240165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000240165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000240165.1</td>\n",
       "      <td>RefSeq_accession</td>\n",
       "      <td>SAMN02603879</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000078850/results/S....</td>\n",
       "      <td>S. cattleya</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000240165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4891</th>\n",
       "      <td>GCA_000286575.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN00255227</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia multivorans CF2</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000286575.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4892</th>\n",
       "      <td>GCA_000959505.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN03144971</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia dolosa AU0158</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000959505.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4893</th>\n",
       "      <td>GCA_000959505.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN03144971</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia dolosa AU0158</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_000959505.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4894</th>\n",
       "      <td>GCA_003568605.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN02866398</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia thailandensis E264</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_003568605.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4895</th>\n",
       "      <td>GCA_003568605.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>SAMN02866398</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...</td>\n",
       "      <td>Burkholderia thailandensis E264</td>\n",
       "      <td>genome</td>\n",
       "      <td>GCA_003568605.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4896 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              NCBI_ID           Database     BioSample  \\\n",
       "0     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "1     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "2     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "3     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "4     GCA_000240165.1   RefSeq_accession  SAMN02603879   \n",
       "...               ...                ...           ...   \n",
       "4891  GCA_000286575.1  GenBank_accession  SAMN00255227   \n",
       "4892  GCA_000959505.1  GenBank_accession  SAMN03144971   \n",
       "4893  GCA_000959505.1  GenBank_accession  SAMN03144971   \n",
       "4894  GCA_003568605.1  GenBank_accession  SAMN02866398   \n",
       "4895  GCA_003568605.1  GenBank_accession  SAMN02866398   \n",
       "\n",
       "                                              LCMS_file  \\\n",
       "0     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "1     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "2     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "3     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "4     ftp://massive.ucsd.edu/MSV000078850/results/S....   \n",
       "...                                                 ...   \n",
       "4891  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4892  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4893  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4894  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "4895  ftp://massive.ucsd.edu/MSV000084945/ccms_peak/...   \n",
       "\n",
       "                         Genome_label  Method      New_NCBI_ID  \n",
       "0                         S. cattleya  genome  GCA_000240165.1  \n",
       "1                         S. cattleya  genome  GCA_000240165.1  \n",
       "2                         S. cattleya  genome  GCA_000240165.1  \n",
       "3                         S. cattleya  genome  GCA_000240165.1  \n",
       "4                         S. cattleya  genome  GCA_000240165.1  \n",
       "...                               ...     ...              ...  \n",
       "4891     Burkholderia multivorans CF2  genome  GCA_000286575.1  \n",
       "4892       Burkholderia dolosa AU0158  genome  GCA_000959505.1  \n",
       "4893       Burkholderia dolosa AU0158  genome  GCA_000959505.1  \n",
       "4894  Burkholderia thailandensis E264  genome  GCA_003568605.1  \n",
       "4895  Burkholderia thailandensis E264  genome  GCA_003568605.1  \n",
       "\n",
       "[4896 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biosample_dict = dict(zip(biosample_df.Accession_id, biosample_df.GenBank_id))\n",
    "new_col = []\n",
    "\n",
    "for i,r in podp_merged_df.iterrows():\n",
    "    biosample_id = podp_merged_df['BioSample'].loc[i]\n",
    "    if biosample_id in biosample_dict.keys():\n",
    "        new_col.append(biosample_dict[biosample_id])\n",
    "    else:\n",
    "        if 'GCA' in biosample_id or 'ERX' in biosample_id or 'ERS' in biosample_id:\n",
    "            new_col.append(biosample_id)\n",
    "        else:\n",
    "            new_col.append('NaN')\n",
    "        \n",
    "podp_merged_df['New_NCBI_ID'] = new_col\n",
    "\n",
    "podp_merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1383 2308 727 4896\n",
      "237\n"
     ]
    }
   ],
   "source": [
    "count = 0\n",
    "gca_list = []\n",
    "\n",
    "for item in podp_merged_df.New_NCBI_ID:\n",
    "    if 'GCA' in item:\n",
    "        count += 1\n",
    "        gca_list.append(item)\n",
    "        \n",
    "print(count,len(podp_merged_df[podp_merged_df['New_NCBI_ID'] == 'NaN']),\n",
    "      len(np.unique(podp_merged_df[podp_merged_df['New_NCBI_ID'] == 'NaN']['NCBI_ID'])),len(podp_merged_df))\n",
    "print(len(np.unique(gca_list)))\n",
    "### notice that all 237 unique BioSamples generate a GCA code and there were 490/727 other genomes that can't be downloaded"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.5. Downloading and unzipping the NCBI genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !mkdir -p /Volumes/TFL190831/NPOmix_round4/podp_ncbi/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "237 0\n"
     ]
    }
   ],
   "source": [
    "cmd_list,genomes_failed = [],[]\n",
    "\n",
    "for i,r in podp_merged_df.iterrows():\n",
    "    if type(r['New_NCBI_ID']) == float:\n",
    "        n = re.match(r'^\\D{4}\\d{2}$',r['NCBI_ID'])\n",
    "        if n:\n",
    "            cmd = 'wget https://sra-download.ncbi.nlm.nih.gov/traces/wgs01/wgs_aux/%s/%s/%s/%s.1.fsa_nt.gz'%(r['NCBI_ID'][0:2],r['NCBI_ID'][2:4],r['NCBI_ID'],r['NCBI_ID'])\n",
    "        if 'GCA' in str(r['NCBI_ID']):\n",
    "            cmd = 'wget -nd -r --no-parent -A \"genomic.fna.gz\"  -R \"*_from_genomic*\" \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/%s/%s/%s/\"'%(r['NCBI_ID'][4:7],r['NCBI_ID'][7:10],r['NCBI_ID'][10:13])\n",
    "        if 'GCA' not in str(r['NCBI_ID']) and not n:\n",
    "            cmd = None\n",
    "    else:\n",
    "        if 'GCA' in str(r['New_NCBI_ID']):\n",
    "            cmd = 'wget -nd -r --no-parent -A \"genomic.fna.gz\"  -R \"*_from_genomic*\" \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/%s/%s/%s/\"'%(r['New_NCBI_ID'][4:7],r['New_NCBI_ID'][7:10],r['New_NCBI_ID'][10:13])\n",
    "        else:\n",
    "            cmd = None\n",
    "    if cmd != None:\n",
    "        if cmd not in cmd_list:\n",
    "            cmd_list.append(cmd)\n",
    "#             try: #only run once\n",
    "#                 subprocess.check_output(cmd, shell=True, cwd='/Volumes/TFL190831/NPOmix_round4/podp_ncbi/')\n",
    "#             except subprocess.CalledProcessError as e:\n",
    "#                 genomes_failed.append(cmd)\n",
    "\n",
    "print(len(cmd_list),len(genomes_failed))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     237\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/TFL190831/NPOmix_round4_genomes/podp_ncbi/ | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "### only run once\n",
    "\n",
    "# genome_list = glob.glob(\"/Volumes/TFL190831/podp_round4_ncbi/*.gz\")\n",
    "\n",
    "# for item in genome_list:\n",
    "#     genome = os.path.basename(item).split('.')[0]\n",
    "#     print(os.path.basename(item))\n",
    "#     cmd = \"gunzip -c %s > %s.fasta\"%(os.path.basename(item),genome)\n",
    "#     subprocess.check_output(cmd, shell=True, cwd='/Volumes/TFL190831/NPOmix_round4_genomes/podp_ncbi/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !rm /Volumes/TFL190831/NPOmix_round4_genomes/podp_ncbi/*.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.6. Copying previously downloaded genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "glob_round3 = glob.glob('/Volumes/TFL190831/podp_round3_ncbi/*fasta')\n",
    "\n",
    "for file_path in glob_round3:\n",
    "    fasta_file = os.path.basename(file_path)\n",
    "    if not os.path.exists('/Volumes/TFL190831/NPOmix_round4_genomes/podp_ncbi/%s'%fasta_file):\n",
    "        cmd = 'cp %s /Volumes/TFL190831/NPOmix_round4/previous_genomes/'%file_path\n",
    "#         subprocess.call(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     237\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/TFL190831/NPOmix_round4_genomes/podp_ncbi/*.fasta | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     174\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/TFL190831/NPOmix_round4_genomes/previous_genomes/*.fasta | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(279, 279)"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AS_previous_list = glob.glob('/Volumes/TFL190831/nf_output-iomega/*')\n",
    "count = 0\n",
    "\n",
    "for file_path in AS_previous_list:\n",
    "    AS_file = os.path.basename(file_path).split('_output')[0]\n",
    "    if os.path.exists('/Volumes/TFL190831/NPOmix_round4_genomes/previous_genomes/%s'%AS_file):\n",
    "        count += 1\n",
    "    else:\n",
    "        if os.path.exists('/Volumes/TFL190831/NPOmix_round4_genomes/podp_ncbi/%s'%AS_file):\n",
    "            count += 1\n",
    "            \n",
    "count,len(AS_previous_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "161"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "previous_genomes = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/previous_genomes/*fasta')\n",
    "missing_from_podp = []\n",
    "\n",
    "for item in previous_genomes:\n",
    "    sum_item = os.path.basename(item).split('.')[0]\n",
    "    temp_df = podp_merged_df[podp_merged_df['NCBI_ID'].str.contains(sum_item)]\n",
    "    if len(temp_df) == 0:\n",
    "        missing_from_podp.append(item)\n",
    "\n",
    "len(missing_from_podp)        \n",
    "### for now, eventhough we don't have these genomes in the current podp_list, we already have LCMS for those"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.7. Downloading metagenomic reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !mkdir /Volumes/TFL190831/NPOmix_round4/ERX_reads/\n",
    "# !mkdir /Volumes/TFL190831/NPOmix_round4/ERS_reads/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "724"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(podp_merged_df[podp_merged_df['New_NCBI_ID'].str.contains('ERS')]['New_NCBI_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "481"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(podp_merged_df[podp_merged_df['New_NCBI_ID'].str.contains('ERX')]['New_NCBI_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "### only run once\n",
    "# !mkdir /Volumes/TFL190831/NPOmix_round4_genomes/ERX_fileroports/\n",
    "# !mkdir /Volumes/TFL190831/NPOmix_round4_genomes/ERS_fileroports/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "### only run once for ERS codes and one more time to ERX\n",
    "\n",
    "# !mkdir /Volumes/TFL190831/NPOmix_round4_genomes/ERS_fileroports/\n",
    "\n",
    "# for i,r in podp_merged_df.iterrows():\n",
    "#     if 'ERS' in podp_merged_df['New_NCBI_ID'].loc[i]:\n",
    "#         ena_code = podp_merged_df['New_NCBI_ID'].loc[i]\n",
    "#         https_cmd = \"wget \\\"https://www.ebi.ac.uk/ena/data/warehouse/filereport?accession=%s&result=read_run&fields=fastq_ftp\\\"\"%ena_code\n",
    "#         subprocess.call(https_cmd, shell=True, cwd='/Volumes/TFL190831/NPOmix_round4_genomes/ERS_fileroports/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "https_list = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/ERX_fileroports/filereport?accession=*')\n",
    "cmd_list = []\n",
    "ena_dict1 = {}\n",
    "\n",
    "for item in https_list:\n",
    "    https_df = pd.read_csv(item,sep='\\t')\n",
    "    read_1 = https_df['fastq_ftp'].loc[0].split(';')[0]\n",
    "    cmd_list.append('wget ftp://%s'%read_1)\n",
    "    ena_dict1[os.path.basename(read_1).split('.')[0]] = item[78:88]\n",
    "    \n",
    "# with open('./temp_files/ftp_ERX_list.txt', 'w') as filehandle:\n",
    "#     filehandle.writelines(\"%s\\n\" % command for command in cmd_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ERX2291668'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ena_dict1['ERR2239508']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "https_list = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/ERS_fileroports/filereport?accession=*')\n",
    "cmd_list = []\n",
    "ena_dict2 = {}\n",
    "\n",
    "for item in https_list:\n",
    "    https_df = pd.read_csv(item,sep='\\t')\n",
    "    read_1 = https_df['fastq_ftp'].loc[0].split(';')[0]\n",
    "    read_2 = https_df['fastq_ftp'].loc[0].split(';')[1]\n",
    "    cmd_list.append('wget ftp://%s'%read_1)\n",
    "    cmd_list.append('wget ftp://%s'%read_2)\n",
    "    ena_dict2[os.path.basename(read_1).split('.')[0].split('_')[0]] = item[78:88]\n",
    "    \n",
    "# with open('./temp_files/ftp_ERS_list.txt', 'w') as filehandle:\n",
    "#     filehandle.writelines(\"%s\\n\" % command for command in cmd_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ERS4346398'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ena_dict2['ERR3946693']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def Merge(dict1, dict2):\n",
    "    res = {**dict1, **dict2}\n",
    "    return res\n",
    "\n",
    "ena_dict3 = Merge(ena_dict1, ena_dict2)\n",
    "\n",
    "# with open('./temp_files/ena_dict-round3-210315.csv', 'w') as f:\n",
    "#     for key in ena_dict3.keys():\n",
    "#         f.write(\"%s,%s\\n\"%(key,ena_dict3[key]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```bash\n",
    "cd /Volumes/TFL190831/NPOmix_round4_genomes/ERX_reads/\n",
    "\n",
    "sh ./ftp_ERX_list.txt\n",
    "```\n",
    "and alternatively\n",
    "\n",
    "```bash\n",
    "cd /Volumes/TFL190831/NPOmix_round4_genomes/ERS_reads/\n",
    "\n",
    "sh ./ftp_ERS_list.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    1448\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/TFL190831/NPOmix_round4_genomes/ERS_reads/*fastq.gz | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     481\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/TFL190831/NPOmix_round4_genomes/ERX_reads/*fastq.gz | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.8. Downloading JGI (meta)genomes\n",
    "\n",
    "`Salinispora_genomes_faa` with 119 Salinisopra was provided by the Jensen lab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "309"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(podp_merged_df[podp_merged_df['Database'].str.contains('JGI')]['NCBI_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !mkdir /Volumes/TFL190831/NPOmix_round4_genomes/manual_genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "download_col,download_list,sal_filt = [],[],[]\n",
    "sal_glob = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/Salinispora_genomes_faa/*.faa')\n",
    "\n",
    "for item in sal_glob:\n",
    "    sal_filt.append(os.path.basename(item).split('_')[0])\n",
    "\n",
    "for i,r in podp_merged_df.iterrows():\n",
    "    if 'JGI' in r['Database']:\n",
    "        if 'Salinispora' in r['Genome_label']:\n",
    "            sal_id = 'S' + r['Genome_label'][12] + r['Genome_label'].split(' ')[2]\n",
    "            if sal_id in sal_filt:\n",
    "                download_col.append('yes')\n",
    "                download_list.append(sal_id)\n",
    "            else:\n",
    "                download_col.append('no')\n",
    "        else:\n",
    "            download_col.append('no')\n",
    "    else:\n",
    "        if r['New_NCBI_ID'] != 'NaN':\n",
    "            download_col.append('yes')\n",
    "        else:\n",
    "            download_col.append('no')\n",
    "            \n",
    "podp_merged_df['Downloaded'] = download_col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(112, 4896, 4896)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(download_list)),len(download_col),len(podp_merged_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1443"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(podp_merged_df[podp_merged_df['Downloaded'] == 'yes']['New_NCBI_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>Database</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>LCMS_file</th>\n",
       "      <th>Genome_label</th>\n",
       "      <th>Method</th>\n",
       "      <th>New_NCBI_ID</th>\n",
       "      <th>Downloaded</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>2821485673</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>SAMN13061032</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000086050/updates/20...</td>\n",
       "      <td>GUM007</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>CP022686.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083387/peak/mzXML...</td>\n",
       "      <td>Ecoli_Nissle_genome</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>CP022686.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083387/peak/mzXML...</td>\n",
       "      <td>Ecoli_Nissle_genome</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>CP022686.1</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083387/peak/mzXML...</td>\n",
       "      <td>Ecoli_Nissle_genome</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>MDRX01000000</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085021/peak/DA2-8...</td>\n",
       "      <td>WMMB235</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3970</th>\n",
       "      <td>651717039, 651716642</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>SAMN11997568</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...</td>\n",
       "      <td>Streptomyces spectabilis Dietz NRRL 2792 (ATCC...</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3971</th>\n",
       "      <td>651717039, 651716642</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>SAMN11997568</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...</td>\n",
       "      <td>Streptomyces spectabilis Dietz NRRL 2792 (ATCC...</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3972</th>\n",
       "      <td>651717039, 651716642</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>SAMN11997568</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...</td>\n",
       "      <td>Streptomyces spectabilis Dietz NRRL 2792 (ATCC...</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3973</th>\n",
       "      <td>2767802005</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>SAMN02645517</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...</td>\n",
       "      <td>Streptomyces sp. NRRL S-325</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4698</th>\n",
       "      <td>235463</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>PRJNA639598</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085586/peak/Actin...</td>\n",
       "      <td>genome - Actinomadura BRA177</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1849 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   NCBI_ID           Database     BioSample  \\\n",
       "46              2821485673      JGI_Genome_ID  SAMN13061032   \n",
       "47              CP022686.1  GenBank_accession  No BioSample   \n",
       "48              CP022686.1  GenBank_accession  No BioSample   \n",
       "49              CP022686.1  GenBank_accession  No BioSample   \n",
       "50            MDRX01000000  GenBank_accession  No BioSample   \n",
       "...                    ...                ...           ...   \n",
       "3970  651717039, 651716642      JGI_Genome_ID  SAMN11997568   \n",
       "3971  651717039, 651716642      JGI_Genome_ID  SAMN11997568   \n",
       "3972  651717039, 651716642      JGI_Genome_ID  SAMN11997568   \n",
       "3973            2767802005      JGI_Genome_ID  SAMN02645517   \n",
       "4698                235463      JGI_Genome_ID   PRJNA639598   \n",
       "\n",
       "                                              LCMS_file  \\\n",
       "46    ftp://massive.ucsd.edu/MSV000086050/updates/20...   \n",
       "47    ftp://massive.ucsd.edu/MSV000083387/peak/mzXML...   \n",
       "48    ftp://massive.ucsd.edu/MSV000083387/peak/mzXML...   \n",
       "49    ftp://massive.ucsd.edu/MSV000083387/peak/mzXML...   \n",
       "50    ftp://massive.ucsd.edu/MSV000085021/peak/DA2-8...   \n",
       "...                                                 ...   \n",
       "3970  ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...   \n",
       "3971  ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...   \n",
       "3972  ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...   \n",
       "3973  ftp://massive.ucsd.edu/MSV000083648/raw/detoxi...   \n",
       "4698  ftp://massive.ucsd.edu/MSV000085586/peak/Actin...   \n",
       "\n",
       "                                           Genome_label  \\\n",
       "46                                               GUM007   \n",
       "47                                  Ecoli_Nissle_genome   \n",
       "48                                  Ecoli_Nissle_genome   \n",
       "49                                  Ecoli_Nissle_genome   \n",
       "50                                              WMMB235   \n",
       "...                                                 ...   \n",
       "3970  Streptomyces spectabilis Dietz NRRL 2792 (ATCC...   \n",
       "3971  Streptomyces spectabilis Dietz NRRL 2792 (ATCC...   \n",
       "3972  Streptomyces spectabilis Dietz NRRL 2792 (ATCC...   \n",
       "3973                        Streptomyces sp. NRRL S-325   \n",
       "4698                       genome - Actinomadura BRA177   \n",
       "\n",
       "                           Method New_NCBI_ID Downloaded  \n",
       "46    metagenome-assembled genome         NaN         no  \n",
       "47                         genome         NaN         no  \n",
       "48                         genome         NaN         no  \n",
       "49                         genome         NaN         no  \n",
       "50                         genome         NaN         no  \n",
       "...                           ...         ...        ...  \n",
       "3970                       genome         NaN         no  \n",
       "3971                       genome         NaN         no  \n",
       "3972                       genome         NaN         no  \n",
       "3973                       genome         NaN         no  \n",
       "4698                       genome         NaN         no  \n",
       "\n",
       "[1849 rows x 8 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "podp_merged_df[podp_merged_df['Downloaded'] == 'no']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use the dataframe above to manually download these genomes searching at [NCBI database](https://www.ncbi.nlm.nih.gov/Traces/wgs/?page=1&view=all&search=GCF_000196155.1) and place them a the manual_genomes folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "manual_glob = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/manual_genomes/*')\n",
    "\n",
    "manual_list,indexes_to_keep = [],[]\n",
    "\n",
    "for item in manual_glob:\n",
    "    manual_list.append(os.path.basename(item).split('.fasta')[0])\n",
    "\n",
    "for i,r in podp_merged_df.iterrows():\n",
    "    if r['Downloaded'] == 'no':\n",
    "        if r['NCBI_ID'] not in manual_list:\n",
    "            indexes_to_keep.append(i)\n",
    "\n",
    "# podp_merged_df.loc[indexes_to_keep].to_csv('./outputs/cant_download_podp_merged_df-round3-210315.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3047, 1849, 4896)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(podp_merged_df[podp_merged_df['Downloaded'] == 'yes']),len(podp_merged_df[podp_merged_df['Downloaded'] == 'no']),len(podp_merged_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "308"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(podp_merged_df.loc[indexes_to_keep]['NCBI_ID']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.9. Separating Genomes without antiSMASH (Optional)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !mkdir /Volumes/TFL190831/genomes_to_antiSMASH/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1766, 679)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AS_all_list = glob.glob('/Volumes/TFL190831/ming_output/antismash/*')\n",
    "AS_list = []\n",
    "count = 0\n",
    "\n",
    "for file_path in AS_all_list:\n",
    "    AS_file = file_path.split('/')[5].split('.')[0].split('_contigs')[0]\n",
    "    if AS_file in ena_dict1:\n",
    "        AS_list.append(ena_dict1[AS_file])\n",
    "    if AS_file in ena_dict2:\n",
    "        AS_list.append(ena_dict2[AS_file])\n",
    "    if AS_file not in ena_dict1 and AS_file not in ena_dict2:\n",
    "        AS_list.append(AS_file)\n",
    "\n",
    "manual_genomes = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/manual_genomes/*')\n",
    "ncbi_genomes = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/podp_ncbi/*')\n",
    "sal_genomes = glob.glob('/Volumes/TFL190831/NPOmix_round4_genomes/Salinispora_genomes_faa/*')\n",
    "\n",
    "for item in list(manual_genomes+ncbi_genomes+sal_genomes):\n",
    "    if os.path.basename(item) not in AS_list:\n",
    "        count += 1\n",
    "        cmd = 'cp %s /Volumes/TFL190831/genomes_to_antiSMASH/'%item\n",
    "#         subprocess.call(cmd,shell=True)\n",
    "    \n",
    "len(AS_list),count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.10. Running antiSMASH\n",
    "\n",
    "```\n",
    "Ming will add code\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.11. Filtering links for (meta)genomes with antiSMASH and downloading LC-MS/MS (need update)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    1766\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/TFL190831/ming_output/antismash/ | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4896\n"
     ]
    }
   ],
   "source": [
    "updated_col = []\n",
    "\n",
    "for i,r in podp_merged_df.iterrows():\n",
    "    if r['New_NCBI_ID'] in AS_list:\n",
    "        updated_col.append('yes')\n",
    "    else:\n",
    "        if r['NCBI_ID'] in AS_list:\n",
    "            updated_col.append('yes')\n",
    "        else:\n",
    "            updated_col.append('no')\n",
    "        \n",
    "print(len(updated_col))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "podp_merged_df['Downloaded'] = updated_col\n",
    "podp_final_df = podp_merged_df[podp_merged_df['Downloaded'] == 'yes']\n",
    "podp_final_df = podp_final_df.reset_index(drop=True)\n",
    "# podp_final_df.to_csv('./outputs/podp_final_df-round4-210426.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>Database</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>LCMS_file</th>\n",
       "      <th>Genome_label</th>\n",
       "      <th>Method</th>\n",
       "      <th>New_NCBI_ID</th>\n",
       "      <th>Downloaded</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2821485673</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>SAMN13061032</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000086050/updates/20...</td>\n",
       "      <td>GUM007</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MDRX01000000</td>\n",
       "      <td>GenBank_accession</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085021/peak/DA2-8...</td>\n",
       "      <td>WMMB235</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2518285561</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000079284//raw/Dunca...</td>\n",
       "      <td>Salinispora pacifica CNR942</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2518285561</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000079284//raw/Dunca...</td>\n",
       "      <td>Salinispora pacifica CNR942</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2518285562</td>\n",
       "      <td>JGI_Genome_ID</td>\n",
       "      <td>No BioSample</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000079284//raw/Dunca...</td>\n",
       "      <td>Salinispora pacifica CNS055</td>\n",
       "      <td>genome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2108</th>\n",
       "      <td>ERS4356320</td>\n",
       "      <td>ENA_NCBI_accession</td>\n",
       "      <td>ERS4356320</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...</td>\n",
       "      <td>ERS4356320</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ERS4356320</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2109</th>\n",
       "      <td>ERS4356321</td>\n",
       "      <td>ENA_NCBI_accession</td>\n",
       "      <td>ERS4356321</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...</td>\n",
       "      <td>ERS4356321</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ERS4356321</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2110</th>\n",
       "      <td>ERS4356322</td>\n",
       "      <td>ENA_NCBI_accession</td>\n",
       "      <td>ERS4356322</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...</td>\n",
       "      <td>ERS4356322</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ERS4356322</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2111</th>\n",
       "      <td>ERS4356323</td>\n",
       "      <td>ENA_NCBI_accession</td>\n",
       "      <td>ERS4356323</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...</td>\n",
       "      <td>ERS4356323</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ERS4356323</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2112</th>\n",
       "      <td>ERS4356324</td>\n",
       "      <td>ENA_NCBI_accession</td>\n",
       "      <td>ERS4356324</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...</td>\n",
       "      <td>ERS4356324</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ERS4356324</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2113 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           NCBI_ID            Database     BioSample  \\\n",
       "0       2821485673       JGI_Genome_ID  SAMN13061032   \n",
       "1     MDRX01000000   GenBank_accession  No BioSample   \n",
       "2       2518285561       JGI_Genome_ID  No BioSample   \n",
       "3       2518285561       JGI_Genome_ID  No BioSample   \n",
       "4       2518285562       JGI_Genome_ID  No BioSample   \n",
       "...            ...                 ...           ...   \n",
       "2108    ERS4356320  ENA_NCBI_accession    ERS4356320   \n",
       "2109    ERS4356321  ENA_NCBI_accession    ERS4356321   \n",
       "2110    ERS4356322  ENA_NCBI_accession    ERS4356322   \n",
       "2111    ERS4356323  ENA_NCBI_accession    ERS4356323   \n",
       "2112    ERS4356324  ENA_NCBI_accession    ERS4356324   \n",
       "\n",
       "                                              LCMS_file  \\\n",
       "0     ftp://massive.ucsd.edu/MSV000086050/updates/20...   \n",
       "1     ftp://massive.ucsd.edu/MSV000085021/peak/DA2-8...   \n",
       "2     ftp://massive.ucsd.edu/MSV000079284//raw/Dunca...   \n",
       "3     ftp://massive.ucsd.edu/MSV000079284//raw/Dunca...   \n",
       "4     ftp://massive.ucsd.edu/MSV000079284//raw/Dunca...   \n",
       "...                                                 ...   \n",
       "2108  ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...   \n",
       "2109  ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...   \n",
       "2110  ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...   \n",
       "2111  ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...   \n",
       "2112  ftp://massive.ucsd.edu/MSV000083302/peak/mzxml...   \n",
       "\n",
       "                     Genome_label                       Method New_NCBI_ID  \\\n",
       "0                          GUM007  metagenome-assembled genome         NaN   \n",
       "1                         WMMB235                       genome         NaN   \n",
       "2     Salinispora pacifica CNR942                       genome         NaN   \n",
       "3     Salinispora pacifica CNR942                       genome         NaN   \n",
       "4     Salinispora pacifica CNS055                       genome         NaN   \n",
       "...                           ...                          ...         ...   \n",
       "2108                   ERS4356320                   metagenome  ERS4356320   \n",
       "2109                   ERS4356321                   metagenome  ERS4356321   \n",
       "2110                   ERS4356322                   metagenome  ERS4356322   \n",
       "2111                   ERS4356323                   metagenome  ERS4356323   \n",
       "2112                   ERS4356324                   metagenome  ERS4356324   \n",
       "\n",
       "     Downloaded  \n",
       "0           yes  \n",
       "1           yes  \n",
       "2           yes  \n",
       "3           yes  \n",
       "4           yes  \n",
       "...         ...  \n",
       "2108        yes  \n",
       "2109        yes  \n",
       "2110        yes  \n",
       "2111        yes  \n",
       "2112        yes  \n",
       "\n",
       "[2113 rows x 8 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "podp_final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((4896, 8), (2113, 8))"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "podp_merged_df.shape,podp_final_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1610"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lcms_list = []\n",
    "\n",
    "lcms_glob = glob.glob('/Volumes/TFL210426/podp_LCMS/*')\n",
    "\n",
    "for lcms_file in lcms_glob:\n",
    "    lcms_list.append(os.path.basename(lcms_file.split('.')[0]))\n",
    "\n",
    "lcms_list = list(np.unique(lcms_list))\n",
    "\n",
    "len(lcms_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows_wo_file,seen,commands = [],[],[]\n",
    "\n",
    "for i,r in podp_final_df.iterrows():\n",
    "    if podp_final_df['New_NCBI_ID'].loc[i] != 'NaN':\n",
    "        genomeID = podp_final_df['New_NCBI_ID'].loc[i]\n",
    "    else:\n",
    "        genomeID = podp_final_df['NCBI_ID'].loc[i]\n",
    "    seen.append(genomeID)\n",
    "    file_count = seen.count(genomeID)\n",
    "    extension = podp_final_df['LCMS_file'].loc[i].rsplit('.', 1)[1]\n",
    "    if genomeID not in lcms_list:\n",
    "        cmd = 'wget -O %s %s'%(genomeID+'.'+extension+'.'+str(file_count),podp_final_df['LCMS_file'].loc[i])\n",
    "        commands.append(cmd)\n",
    "\n",
    "len(commands)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "### only run once\n",
    "\n",
    "# with open('./temp_files/PODP_LCMS_list_commands-round4.txt', \"w\") as output:\n",
    "#     writer = csv.writer(output, lineterminator='\\n')\n",
    "#     for val in commands:\n",
    "#         writer.writerow([val])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```bash\n",
    "mkdir /Volumes/TFL190831/podp_LCMS\n",
    "\n",
    "cd /Volumes/TFL190831/podp_LCMS\n",
    "\n",
    "sh PODP_LCMS_list_commands-round3.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Rename names containing \".1.\"**\n",
    "\n",
    "```\n",
    "cd podp_LCMS/\n",
    "\n",
    "ls *.1.* | cat > rename.txt\n",
    "\n",
    "Edit rename.txt in a text editor\n",
    "\n",
    "sh rename.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.12 Download missing Gerwick cyanoibacterial MS/MS and selecting Gerwick mgf files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pyteomics import mgf, auxiliary\n",
    "import csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "Download all GNPS database from the link: https://gnps.ucsd.edu/ProteoSAFe/gnpslibrary.jsp?library=all\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "448698 448698\n"
     ]
    }
   ],
   "source": [
    "count = 0\n",
    "colA = []\n",
    "colB = []\n",
    "\n",
    "with open('/Users/tiagoferreiraleao/Dropbox/tiago-NAS/NPOmix/inputs/ALL_GNPS.mgf') as fp:\n",
    "    for line in fp:\n",
    "        count += 1\n",
    "        a = re.match(r'BEGIN IONS',line)\n",
    "        if a:\n",
    "            colA.append(count)\n",
    "        b = re.match(r'END IONS',line)\n",
    "        if b:\n",
    "            colB.append(count)\n",
    "\n",
    "print(len(colA),len(colB))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(448698, 448698, 448698, 448698, 448698)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col1 = []\n",
    "col2 = []\n",
    "col3 = []\n",
    "col4 = []\n",
    "col5 = []\n",
    "\n",
    "with open('/Users/tiagoferreiraleao/Dropbox/tiago-NAS/NPOmix/inputs/ALL_GNPS.mgf') as fp:\n",
    "    lines=fp.readlines()\n",
    "    for index,item in enumerate(colA):\n",
    "        pepmass = \"N/A\"\n",
    "        name = \"N/A\"\n",
    "        pi = \"N/A\"\n",
    "        smiles = \"N/A\"\n",
    "        gnps_id = \"N/A\"\n",
    "        chunk = lines[item:colB[index]]\n",
    "        for line in chunk:\n",
    "            a = re.match(r'PEPMASS=(.*)',line)\n",
    "            if a:\n",
    "                pepmass = a.group(1)\n",
    "            b = re.match(r'NAME=(.*)',line)\n",
    "            if b:\n",
    "                name = b.group(1)\n",
    "            c = re.match(r'PI=(.*)',line)\n",
    "            if c:\n",
    "                pi = c.group(1)\n",
    "            d = re.match(r'SMILES=(.*)',line)\n",
    "            if d:\n",
    "                smiles = d.group(1)\n",
    "            e = re.match(r'SPECTRUMID=(.*)',line)\n",
    "            if e:\n",
    "                gnps_id = e.group(1)\n",
    "        col1.append(pepmass)\n",
    "        col2.append(name)\n",
    "        col3.append(pi)\n",
    "        col4.append(smiles)\n",
    "        col5.append(gnps_id)\n",
    "                \n",
    "len(col1),len(col2),len(col3),len(col4),len(col5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PEPMASS</th>\n",
       "      <th>NAME</th>\n",
       "      <th>PI</th>\n",
       "      <th>SMILES</th>\n",
       "      <th>SPECTRUMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>981.54</td>\n",
       "      <td>3-Des-Microcystein_LR M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CC(C)CC1NC(=O)C(C)NC(=O)C(=C)N(C)C(=O)CCC(NC(=...</td>\n",
       "      <td>CCMSLIB00000001547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>940.25</td>\n",
       "      <td>Hoiamide B M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCC[C@@H](C)[C@@H]([C@H](C)[C@@H]1[C@H]([C@H](...</td>\n",
       "      <td>CCMSLIB00000001548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>456.1</td>\n",
       "      <td>Malyngamide C M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCCCCCC[C@@H](C/C=C/CCC(=O)NC/C(=C/Cl)/[C@@]12...</td>\n",
       "      <td>CCMSLIB00000001549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>545.0</td>\n",
       "      <td>Scytonemin M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>OC1=CC=C(\\C=C2\\C(=O)C(C3=C4C5=C(C=CC=C5)N=C4\\C...</td>\n",
       "      <td>CCMSLIB00000001550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>314.116</td>\n",
       "      <td>Salinisporamide A M+H</td>\n",
       "      <td>Fenical-Jensen-Moore</td>\n",
       "      <td>N/A</td>\n",
       "      <td>CCMSLIB00000001551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448693</th>\n",
       "      <td>277.218</td>\n",
       "      <td>Pinolenic acid - 20eV M-H</td>\n",
       "      <td>Sumner</td>\n",
       "      <td>CCCCC/C=C\\C/C=C\\CC/C=C\\CCCC(=O)O</td>\n",
       "      <td>CCMSLIB00004684135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448694</th>\n",
       "      <td>277.218</td>\n",
       "      <td>Pinolenic acid - 30eV M-H</td>\n",
       "      <td>Sumner</td>\n",
       "      <td>CCCCC/C=C\\C/C=C\\CC/C=C\\CCCC(=O)O</td>\n",
       "      <td>CCMSLIB00004684136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448695</th>\n",
       "      <td>279.234</td>\n",
       "      <td>9Z, 11E-Linoleic acid - 10eV M-H</td>\n",
       "      <td>Sumner</td>\n",
       "      <td>CCCCCC/C=C/C=C\\CCCCCCCC(=O)O</td>\n",
       "      <td>CCMSLIB00004684137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448696</th>\n",
       "      <td>279.232</td>\n",
       "      <td>9Z, 11E-Linoleic acid - 30eV M-H</td>\n",
       "      <td>Sumner</td>\n",
       "      <td>CCCCCC/C=C/C=C\\CCCCCCCC(=O)O</td>\n",
       "      <td>CCMSLIB00004684138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448697</th>\n",
       "      <td>279.233</td>\n",
       "      <td>9Z, 11E-Linoleic acid - 20eV M-H</td>\n",
       "      <td>Sumner</td>\n",
       "      <td>CCCCCC/C=C/C=C\\CCCCCCCC(=O)O</td>\n",
       "      <td>CCMSLIB00004684139</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>448698 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PEPMASS                              NAME                    PI  \\\n",
       "0        981.54         3-Des-Microcystein_LR M+H               Gerwick   \n",
       "1        940.25                    Hoiamide B M+H               Gerwick   \n",
       "2         456.1                 Malyngamide C M+H               Gerwick   \n",
       "3         545.0                    Scytonemin M+H               Gerwick   \n",
       "4       314.116             Salinisporamide A M+H  Fenical-Jensen-Moore   \n",
       "...         ...                               ...                   ...   \n",
       "448693  277.218         Pinolenic acid - 20eV M-H                Sumner   \n",
       "448694  277.218         Pinolenic acid - 30eV M-H                Sumner   \n",
       "448695  279.234  9Z, 11E-Linoleic acid - 10eV M-H                Sumner   \n",
       "448696  279.232  9Z, 11E-Linoleic acid - 30eV M-H                Sumner   \n",
       "448697  279.233  9Z, 11E-Linoleic acid - 20eV M-H                Sumner   \n",
       "\n",
       "                                                   SMILES          SPECTRUMID  \n",
       "0       CC(C)CC1NC(=O)C(C)NC(=O)C(=C)N(C)C(=O)CCC(NC(=...  CCMSLIB00000001547  \n",
       "1       CCC[C@@H](C)[C@@H]([C@H](C)[C@@H]1[C@H]([C@H](...  CCMSLIB00000001548  \n",
       "2       CCCCCCC[C@@H](C/C=C/CCC(=O)NC/C(=C/Cl)/[C@@]12...  CCMSLIB00000001549  \n",
       "3       OC1=CC=C(\\C=C2\\C(=O)C(C3=C4C5=C(C=CC=C5)N=C4\\C...  CCMSLIB00000001550  \n",
       "4                                                     N/A  CCMSLIB00000001551  \n",
       "...                                                   ...                 ...  \n",
       "448693                   CCCCC/C=C\\C/C=C\\CC/C=C\\CCCC(=O)O  CCMSLIB00004684135  \n",
       "448694                   CCCCC/C=C\\C/C=C\\CC/C=C\\CCCC(=O)O  CCMSLIB00004684136  \n",
       "448695                       CCCCCC/C=C/C=C\\CCCCCCCC(=O)O  CCMSLIB00004684137  \n",
       "448696                       CCCCCC/C=C/C=C\\CCCCCCCC(=O)O  CCMSLIB00004684138  \n",
       "448697                       CCCCCC/C=C/C=C\\CCCCCCCC(=O)O  CCMSLIB00004684139  \n",
       "\n",
       "[448698 rows x 5 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnps_df = pd.DataFrame(\n",
    "    {'PEPMASS': col1,\n",
    "     'NAME': col2,\n",
    "     'PI': col3,\n",
    "     'SMILES': col4,\n",
    "     'SPECTRUMID': col5\n",
    "    })\n",
    "\n",
    "gnps_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PEPMASS</th>\n",
       "      <th>NAME</th>\n",
       "      <th>PI</th>\n",
       "      <th>SMILES</th>\n",
       "      <th>SPECTRUMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>981.54</td>\n",
       "      <td>3-Des-Microcystein_LR M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CC(C)CC1NC(=O)C(C)NC(=O)C(=C)N(C)C(=O)CCC(NC(=...</td>\n",
       "      <td>CCMSLIB00000001547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>940.25</td>\n",
       "      <td>Hoiamide B M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCC[C@@H](C)[C@@H]([C@H](C)[C@@H]1[C@H]([C@H](...</td>\n",
       "      <td>CCMSLIB00000001548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>456.1</td>\n",
       "      <td>Malyngamide C M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCCCCCC[C@@H](C/C=C/CCC(=O)NC/C(=C/Cl)/[C@@]12...</td>\n",
       "      <td>CCMSLIB00000001549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>545.0</td>\n",
       "      <td>Scytonemin M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>OC1=CC=C(\\C=C2\\C(=O)C(C3=C4C5=C(C=CC=C5)N=C4\\C...</td>\n",
       "      <td>CCMSLIB00000001550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>667.115</td>\n",
       "      <td>Hectochlorin M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>C[C@H]1[C@@H](OC(C2=CSC([C@H](C(C)(OC(C3=CSC([...</td>\n",
       "      <td>CCMSLIB00000001552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6142</th>\n",
       "      <td>1201.5</td>\n",
       "      <td>Kurahamide B M+NH4</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCCC(N[C@H](C(N[C@H](C(N[C@H]1[C@@H](C)OC([C@H...</td>\n",
       "      <td>CCMSLIB00005723420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6702</th>\n",
       "      <td>1156.59</td>\n",
       "      <td>Rivulariapeptolide 1155 M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CC(C)[C@@H](NC([C@@H](CC1=CC=C(O)C=C1)N(C)C([C...</td>\n",
       "      <td>CCMSLIB00005723986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6714</th>\n",
       "      <td>400.25</td>\n",
       "      <td>Cryptomaldamice M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>NC(N[C@@H](CO)C(N[C@@H](C(C)C)C(N(C)[C@@H](C(C...</td>\n",
       "      <td>CCMSLIB00005724004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6726</th>\n",
       "      <td>400.257</td>\n",
       "      <td>Cryptomaldamide M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CC(C)[C@@H](/C=C(\\C)/C(=O)O)N(C)C(=O)[C@H](C(C...</td>\n",
       "      <td>CCMSLIB00005724017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6771</th>\n",
       "      <td>1141.47</td>\n",
       "      <td>yessotoxin M-H</td>\n",
       "      <td>W. Gerwick</td>\n",
       "      <td>C[C@H]1CC[C@]2([C@@H](C[C@H]3[C@](O2)(CC[C@H]4...</td>\n",
       "      <td>CCMSLIB00005724062</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>629 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      PEPMASS                         NAME          PI  \\\n",
       "0      981.54    3-Des-Microcystein_LR M+H     Gerwick   \n",
       "1      940.25               Hoiamide B M+H     Gerwick   \n",
       "2       456.1            Malyngamide C M+H     Gerwick   \n",
       "3       545.0               Scytonemin M+H     Gerwick   \n",
       "5     667.115             Hectochlorin M+H     Gerwick   \n",
       "...       ...                          ...         ...   \n",
       "6142   1201.5           Kurahamide B M+NH4     Gerwick   \n",
       "6702  1156.59  Rivulariapeptolide 1155 M+H     Gerwick   \n",
       "6714   400.25          Cryptomaldamice M+H     Gerwick   \n",
       "6726  400.257          Cryptomaldamide M+H     Gerwick   \n",
       "6771  1141.47               yessotoxin M-H  W. Gerwick   \n",
       "\n",
       "                                                 SMILES          SPECTRUMID  \n",
       "0     CC(C)CC1NC(=O)C(C)NC(=O)C(=C)N(C)C(=O)CCC(NC(=...  CCMSLIB00000001547  \n",
       "1     CCC[C@@H](C)[C@@H]([C@H](C)[C@@H]1[C@H]([C@H](...  CCMSLIB00000001548  \n",
       "2     CCCCCCC[C@@H](C/C=C/CCC(=O)NC/C(=C/Cl)/[C@@]12...  CCMSLIB00000001549  \n",
       "3     OC1=CC=C(\\C=C2\\C(=O)C(C3=C4C5=C(C=CC=C5)N=C4\\C...  CCMSLIB00000001550  \n",
       "5     C[C@H]1[C@@H](OC(C2=CSC([C@H](C(C)(OC(C3=CSC([...  CCMSLIB00000001552  \n",
       "...                                                 ...                 ...  \n",
       "6142  CCCC(N[C@H](C(N[C@H](C(N[C@H]1[C@@H](C)OC([C@H...  CCMSLIB00005723420  \n",
       "6702  CC(C)[C@@H](NC([C@@H](CC1=CC=C(O)C=C1)N(C)C([C...  CCMSLIB00005723986  \n",
       "6714  NC(N[C@@H](CO)C(N[C@@H](C(C)C)C(N(C)[C@@H](C(C...  CCMSLIB00005724004  \n",
       "6726  CC(C)[C@@H](/C=C(\\C)/C(=O)O)N(C)C(=O)[C@H](C(C...  CCMSLIB00005724017  \n",
       "6771  C[C@H]1CC[C@]2([C@@H](C[C@H]3[C@](O2)(CC[C@H]4...  CCMSLIB00005724062  \n",
       "\n",
       "[629 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gerwick_df = gnps_df[gnps_df[\"PI\"].isin([\"Gerwick\",\"Dr. Gerwick\",\"W. Gerwick\",\"Dorrestein/Gerwick\",\"Pieter Dorrestein &amp; Lena Gerwick\",\"William Gerwick\"])]\n",
    "\n",
    "gerwick_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PEPMASS</th>\n",
       "      <th>NAME</th>\n",
       "      <th>PI</th>\n",
       "      <th>SMILES</th>\n",
       "      <th>SPECTRUMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>981.54</td>\n",
       "      <td>3-Des-Microcystein_LR M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CC(C)CC1NC(=O)C(C)NC(=O)C(=C)N(C)C(=O)CCC(NC(=...</td>\n",
       "      <td>CCMSLIB00000001547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>940.25</td>\n",
       "      <td>Hoiamide B M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCC[C@@H](C)[C@@H]([C@H](C)[C@@H]1[C@H]([C@H](...</td>\n",
       "      <td>CCMSLIB00000001548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>456.1</td>\n",
       "      <td>Malyngamide C M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCCCCCC[C@@H](C/C=C/CCC(=O)NC/C(=C/Cl)/[C@@]12...</td>\n",
       "      <td>CCMSLIB00000001549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>545.0</td>\n",
       "      <td>Scytonemin M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>OC1=CC=C(\\C=C2\\C(=O)C(C3=C4C5=C(C=CC=C5)N=C4\\C...</td>\n",
       "      <td>CCMSLIB00000001550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>667.115</td>\n",
       "      <td>Hectochlorin M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>C[C@H]1[C@@H](OC(C2=CSC([C@H](C(C)(OC(C3=CSC([...</td>\n",
       "      <td>CCMSLIB00000001552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401</th>\n",
       "      <td>1201.5</td>\n",
       "      <td>Kurahamide B M+NH4</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CCCC(N[C@H](C(N[C@H](C(N[C@H]1[C@@H](C)OC([C@H...</td>\n",
       "      <td>CCMSLIB00005723420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>1156.59</td>\n",
       "      <td>Rivulariapeptolide 1155 M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CC(C)[C@@H](NC([C@@H](CC1=CC=C(O)C=C1)N(C)C([C...</td>\n",
       "      <td>CCMSLIB00005723986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>400.25</td>\n",
       "      <td>Cryptomaldamice M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>NC(N[C@@H](CO)C(N[C@@H](C(C)C)C(N(C)[C@@H](C(C...</td>\n",
       "      <td>CCMSLIB00005724004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>400.257</td>\n",
       "      <td>Cryptomaldamide M+H</td>\n",
       "      <td>Gerwick</td>\n",
       "      <td>CC(C)[C@@H](/C=C(\\C)/C(=O)O)N(C)C(=O)[C@H](C(C...</td>\n",
       "      <td>CCMSLIB00005724017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>1141.47</td>\n",
       "      <td>yessotoxin M-H</td>\n",
       "      <td>W. Gerwick</td>\n",
       "      <td>C[C@H]1CC[C@]2([C@@H](C[C@H]3[C@](O2)(CC[C@H]4...</td>\n",
       "      <td>CCMSLIB00005724062</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>406 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     PEPMASS                         NAME          PI  \\\n",
       "0     981.54    3-Des-Microcystein_LR M+H     Gerwick   \n",
       "1     940.25               Hoiamide B M+H     Gerwick   \n",
       "2      456.1            Malyngamide C M+H     Gerwick   \n",
       "3      545.0               Scytonemin M+H     Gerwick   \n",
       "4    667.115             Hectochlorin M+H     Gerwick   \n",
       "..       ...                          ...         ...   \n",
       "401   1201.5           Kurahamide B M+NH4     Gerwick   \n",
       "402  1156.59  Rivulariapeptolide 1155 M+H     Gerwick   \n",
       "403   400.25          Cryptomaldamice M+H     Gerwick   \n",
       "404  400.257          Cryptomaldamide M+H     Gerwick   \n",
       "405  1141.47               yessotoxin M-H  W. Gerwick   \n",
       "\n",
       "                                                SMILES          SPECTRUMID  \n",
       "0    CC(C)CC1NC(=O)C(C)NC(=O)C(=C)N(C)C(=O)CCC(NC(=...  CCMSLIB00000001547  \n",
       "1    CCC[C@@H](C)[C@@H]([C@H](C)[C@@H]1[C@H]([C@H](...  CCMSLIB00000001548  \n",
       "2    CCCCCCC[C@@H](C/C=C/CCC(=O)NC/C(=C/Cl)/[C@@]12...  CCMSLIB00000001549  \n",
       "3    OC1=CC=C(\\C=C2\\C(=O)C(C3=C4C5=C(C=CC=C5)N=C4\\C...  CCMSLIB00000001550  \n",
       "4    C[C@H]1[C@@H](OC(C2=CSC([C@H](C(C)(OC(C3=CSC([...  CCMSLIB00000001552  \n",
       "..                                                 ...                 ...  \n",
       "401  CCCC(N[C@H](C(N[C@H](C(N[C@H]1[C@@H](C)OC([C@H...  CCMSLIB00005723420  \n",
       "402  CC(C)[C@@H](NC([C@@H](CC1=CC=C(O)C=C1)N(C)C([C...  CCMSLIB00005723986  \n",
       "403  NC(N[C@@H](CO)C(N[C@@H](C(C)C)C(N(C)[C@@H](C(C...  CCMSLIB00005724004  \n",
       "404  CC(C)[C@@H](/C=C(\\C)/C(=O)O)N(C)C(=O)[C@H](C(C...  CCMSLIB00005724017  \n",
       "405  C[C@H]1CC[C@]2([C@@H](C[C@H]3[C@](O2)(CC[C@H]4...  CCMSLIB00005724062  \n",
       "\n",
       "[406 rows x 5 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnps_df2 = gerwick_df[gerwick_df[\"SMILES\"] != \"N/A\"]\n",
    "gnps_df2 = gnps_df2[gnps_df2[\"SMILES\"] != ' ']\n",
    "gnps_df2 = gnps_df2[gnps_df2[\"SMILES\"] != 'NaN']\n",
    "gnps_df2 = gnps_df2.fillna(0)\n",
    "gnps_df2 = gnps_df2[gnps_df2[\"SMILES\"] != 0]\n",
    "gnps_df2 = gnps_df2[gnps_df2[\"SMILES\"] != 'c']\n",
    "gnps_df2 = gnps_df2[gnps_df2[\"SMILES\"] != 'no data']\n",
    "\n",
    "gnps_df2.reset_index(inplace=True,drop=True)\n",
    "\n",
    "gnps_df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CCMSLIB00000001547',\n",
       " 'CCMSLIB00000001548',\n",
       " 'CCMSLIB00000001549',\n",
       " 'CCMSLIB00000001550',\n",
       " 'CCMSLIB00000001552',\n",
       " 'CCMSLIB00000001553',\n",
       " 'CCMSLIB00000001555',\n",
       " 'CCMSLIB00000001558',\n",
       " 'CCMSLIB00000001560',\n",
       " 'CCMSLIB00000001562',\n",
       " 'CCMSLIB00000001564',\n",
       " 'CCMSLIB00000001565',\n",
       " 'CCMSLIB00000001567',\n",
       " 'CCMSLIB00000001569',\n",
       " 'CCMSLIB00000001571',\n",
       " 'CCMSLIB00000001573',\n",
       " 'CCMSLIB00000001575',\n",
       " 'CCMSLIB00000001577',\n",
       " 'CCMSLIB00000001579',\n",
       " 'CCMSLIB00000001581',\n",
       " 'CCMSLIB00000001583',\n",
       " 'CCMSLIB00000001585',\n",
       " 'CCMSLIB00000001587',\n",
       " 'CCMSLIB00000001589',\n",
       " 'CCMSLIB00000001591',\n",
       " 'CCMSLIB00000001595',\n",
       " 'CCMSLIB00000001597',\n",
       " 'CCMSLIB00000001601',\n",
       " 'CCMSLIB00000001603',\n",
       " 'CCMSLIB00000001605',\n",
       " 'CCMSLIB00000001608',\n",
       " 'CCMSLIB00000001626',\n",
       " 'CCMSLIB00000001628',\n",
       " 'CCMSLIB00000001630',\n",
       " 'CCMSLIB00000001632',\n",
       " 'CCMSLIB00000001634',\n",
       " 'CCMSLIB00000001636',\n",
       " 'CCMSLIB00000001638',\n",
       " 'CCMSLIB00000001640',\n",
       " 'CCMSLIB00000001642',\n",
       " 'CCMSLIB00000001644',\n",
       " 'CCMSLIB00000001646',\n",
       " 'CCMSLIB00000001648',\n",
       " 'CCMSLIB00000001650',\n",
       " 'CCMSLIB00000001652',\n",
       " 'CCMSLIB00000001654',\n",
       " 'CCMSLIB00000001656',\n",
       " 'CCMSLIB00000001658',\n",
       " 'CCMSLIB00000001660',\n",
       " 'CCMSLIB00000001662',\n",
       " 'CCMSLIB00000001664',\n",
       " 'CCMSLIB00000001666',\n",
       " 'CCMSLIB00000001674',\n",
       " 'CCMSLIB00000001675',\n",
       " 'CCMSLIB00000001677',\n",
       " 'CCMSLIB00000001679',\n",
       " 'CCMSLIB00000001683',\n",
       " 'CCMSLIB00000001685',\n",
       " 'CCMSLIB00000001687',\n",
       " 'CCMSLIB00000001688',\n",
       " 'CCMSLIB00000001689',\n",
       " 'CCMSLIB00000001690',\n",
       " 'CCMSLIB00000001691',\n",
       " 'CCMSLIB00000001692',\n",
       " 'CCMSLIB00000001693',\n",
       " 'CCMSLIB00000001694',\n",
       " 'CCMSLIB00000001695',\n",
       " 'CCMSLIB00000001697',\n",
       " 'CCMSLIB00000001698',\n",
       " 'CCMSLIB00000001699',\n",
       " 'CCMSLIB00000001700',\n",
       " 'CCMSLIB00000001701',\n",
       " 'CCMSLIB00000001702',\n",
       " 'CCMSLIB00000001703',\n",
       " 'CCMSLIB00000001704',\n",
       " 'CCMSLIB00000001705',\n",
       " 'CCMSLIB00000001706',\n",
       " 'CCMSLIB00000001707',\n",
       " 'CCMSLIB00000001708',\n",
       " 'CCMSLIB00000001711',\n",
       " 'CCMSLIB00000001712',\n",
       " 'CCMSLIB00000001713',\n",
       " 'CCMSLIB00000001714',\n",
       " 'CCMSLIB00000001715',\n",
       " 'CCMSLIB00000001716',\n",
       " 'CCMSLIB00000001717',\n",
       " 'CCMSLIB00000001718',\n",
       " 'CCMSLIB00000001719',\n",
       " 'CCMSLIB00000001720',\n",
       " 'CCMSLIB00000001721',\n",
       " 'CCMSLIB00000001722',\n",
       " 'CCMSLIB00000001723',\n",
       " 'CCMSLIB00000001724',\n",
       " 'CCMSLIB00000001727',\n",
       " 'CCMSLIB00000001728',\n",
       " 'CCMSLIB00000001729',\n",
       " 'CCMSLIB00000001730',\n",
       " 'CCMSLIB00000001731',\n",
       " 'CCMSLIB00000001732',\n",
       " 'CCMSLIB00000001733',\n",
       " 'CCMSLIB00000001742',\n",
       " 'CCMSLIB00000001743',\n",
       " 'CCMSLIB00000001744',\n",
       " 'CCMSLIB00000001747',\n",
       " 'CCMSLIB00000001751',\n",
       " 'CCMSLIB00000001756',\n",
       " 'CCMSLIB00000001757',\n",
       " 'CCMSLIB00000001758',\n",
       " 'CCMSLIB00000001759',\n",
       " 'CCMSLIB00000001760',\n",
       " 'CCMSLIB00000001761',\n",
       " 'CCMSLIB00000001762',\n",
       " 'CCMSLIB00000001763',\n",
       " 'CCMSLIB00000001768',\n",
       " 'CCMSLIB00000001769',\n",
       " 'CCMSLIB00000001770',\n",
       " 'CCMSLIB00000001771',\n",
       " 'CCMSLIB00000001772',\n",
       " 'CCMSLIB00000001773',\n",
       " 'CCMSLIB00000001774',\n",
       " 'CCMSLIB00000001776',\n",
       " 'CCMSLIB00000001777',\n",
       " 'CCMSLIB00000001778',\n",
       " 'CCMSLIB00000001783',\n",
       " 'CCMSLIB00000001784',\n",
       " 'CCMSLIB00000001785',\n",
       " 'CCMSLIB00000001786',\n",
       " 'CCMSLIB00000001787',\n",
       " 'CCMSLIB00000001788',\n",
       " 'CCMSLIB00000001789',\n",
       " 'CCMSLIB00000001790',\n",
       " 'CCMSLIB00000001791',\n",
       " 'CCMSLIB00000001792',\n",
       " 'CCMSLIB00000001793',\n",
       " 'CCMSLIB00000001797',\n",
       " 'CCMSLIB00000001798',\n",
       " 'CCMSLIB00000001799',\n",
       " 'CCMSLIB00000001800',\n",
       " 'CCMSLIB00000001803',\n",
       " 'CCMSLIB00000001804',\n",
       " 'CCMSLIB00000001805',\n",
       " 'CCMSLIB00000001806',\n",
       " 'CCMSLIB00000001809',\n",
       " 'CCMSLIB00000001810',\n",
       " 'CCMSLIB00000001811',\n",
       " 'CCMSLIB00000001812',\n",
       " 'CCMSLIB00000001813',\n",
       " 'CCMSLIB00000001814',\n",
       " 'CCMSLIB00000001815',\n",
       " 'CCMSLIB00000001817',\n",
       " 'CCMSLIB00000006872',\n",
       " 'CCMSLIB00000006873',\n",
       " 'CCMSLIB00000006874',\n",
       " 'CCMSLIB00000006875',\n",
       " 'CCMSLIB00000006883',\n",
       " 'CCMSLIB00000006900',\n",
       " 'CCMSLIB00000068215',\n",
       " 'CCMSLIB00000068216',\n",
       " 'CCMSLIB00000068217',\n",
       " 'CCMSLIB00000068218',\n",
       " 'CCMSLIB00000068219',\n",
       " 'CCMSLIB00000068221',\n",
       " 'CCMSLIB00000068223',\n",
       " 'CCMSLIB00000068226',\n",
       " 'CCMSLIB00000068229',\n",
       " 'CCMSLIB00000068232',\n",
       " 'CCMSLIB00000070245',\n",
       " 'CCMSLIB00000070246',\n",
       " 'CCMSLIB00000070247',\n",
       " 'CCMSLIB00000070255',\n",
       " 'CCMSLIB00000070256',\n",
       " 'CCMSLIB00000070257',\n",
       " 'CCMSLIB00000070258',\n",
       " 'CCMSLIB00000070259',\n",
       " 'CCMSLIB00000070260',\n",
       " 'CCMSLIB00000070261',\n",
       " 'CCMSLIB00000070263',\n",
       " 'CCMSLIB00000070265',\n",
       " 'CCMSLIB00000070266',\n",
       " 'CCMSLIB00000070282',\n",
       " 'CCMSLIB00000070287',\n",
       " 'CCMSLIB00000070288',\n",
       " 'CCMSLIB00000070289',\n",
       " 'CCMSLIB00000070290',\n",
       " 'CCMSLIB00000070293',\n",
       " 'CCMSLIB00000070294',\n",
       " 'CCMSLIB00000070295',\n",
       " 'CCMSLIB00000070296',\n",
       " 'CCMSLIB00000070301',\n",
       " 'CCMSLIB00000070302',\n",
       " 'CCMSLIB00000070304',\n",
       " 'CCMSLIB00000071748',\n",
       " 'CCMSLIB00000071750',\n",
       " 'CCMSLIB00000071751',\n",
       " 'CCMSLIB00000071752',\n",
       " 'CCMSLIB00000071754',\n",
       " 'CCMSLIB00000071755',\n",
       " 'CCMSLIB00000071756',\n",
       " 'CCMSLIB00000071757',\n",
       " 'CCMSLIB00000071761',\n",
       " 'CCMSLIB00000071762',\n",
       " 'CCMSLIB00000072478',\n",
       " 'CCMSLIB00000072479',\n",
       " 'CCMSLIB00000072480',\n",
       " 'CCMSLIB00000072481',\n",
       " 'CCMSLIB00000072489',\n",
       " 'CCMSLIB00000072490',\n",
       " 'CCMSLIB00000072491',\n",
       " 'CCMSLIB00000072492',\n",
       " 'CCMSLIB00000072493',\n",
       " 'CCMSLIB00000072494',\n",
       " 'CCMSLIB00000072501',\n",
       " 'CCMSLIB00000072502',\n",
       " 'CCMSLIB00000072503',\n",
       " 'CCMSLIB00000072504',\n",
       " 'CCMSLIB00000072507',\n",
       " 'CCMSLIB00000072508',\n",
       " 'CCMSLIB00000072509',\n",
       " 'CCMSLIB00000072510',\n",
       " 'CCMSLIB00000072511',\n",
       " 'CCMSLIB00000072512',\n",
       " 'CCMSLIB00000072521',\n",
       " 'CCMSLIB00000072522',\n",
       " 'CCMSLIB00000072529',\n",
       " 'CCMSLIB00000072530',\n",
       " 'CCMSLIB00000078897',\n",
       " 'CCMSLIB00000078898',\n",
       " 'CCMSLIB00000424800',\n",
       " 'CCMSLIB00000424822',\n",
       " 'CCMSLIB00000424823',\n",
       " 'CCMSLIB00000424827',\n",
       " 'CCMSLIB00000424828',\n",
       " 'CCMSLIB00000477991',\n",
       " 'CCMSLIB00000477995',\n",
       " 'CCMSLIB00000478597',\n",
       " 'CCMSLIB00000478598',\n",
       " 'CCMSLIB00000478599',\n",
       " 'CCMSLIB00000478600',\n",
       " 'CCMSLIB00000478601',\n",
       " 'CCMSLIB00000478602',\n",
       " 'CCMSLIB00000478603',\n",
       " 'CCMSLIB00000531496',\n",
       " 'CCMSLIB00000531500',\n",
       " 'CCMSLIB00000531501',\n",
       " 'CCMSLIB00000531502',\n",
       " 'CCMSLIB00000840559',\n",
       " 'CCMSLIB00000840560',\n",
       " 'CCMSLIB00000840561',\n",
       " 'CCMSLIB00004682331',\n",
       " 'CCMSLIB00004722207',\n",
       " 'CCMSLIB00004722208',\n",
       " 'CCMSLIB00004722209',\n",
       " 'CCMSLIB00004722210',\n",
       " 'CCMSLIB00004722211',\n",
       " 'CCMSLIB00004722212',\n",
       " 'CCMSLIB00004722213',\n",
       " 'CCMSLIB00004722214',\n",
       " 'CCMSLIB00004722215',\n",
       " 'CCMSLIB00004722216',\n",
       " 'CCMSLIB00004722217',\n",
       " 'CCMSLIB00004722218',\n",
       " 'CCMSLIB00004722219',\n",
       " 'CCMSLIB00004722220',\n",
       " 'CCMSLIB00004722221',\n",
       " 'CCMSLIB00004722222',\n",
       " 'CCMSLIB00004722223',\n",
       " 'CCMSLIB00004752995',\n",
       " 'CCMSLIB00004752996',\n",
       " 'CCMSLIB00004753000',\n",
       " 'CCMSLIB00004753001',\n",
       " 'CCMSLIB00004753002',\n",
       " 'CCMSLIB00004753003',\n",
       " 'CCMSLIB00004753004',\n",
       " 'CCMSLIB00005435621',\n",
       " 'CCMSLIB00005435622',\n",
       " 'CCMSLIB00005435792',\n",
       " 'CCMSLIB00005436029',\n",
       " 'CCMSLIB00005436030',\n",
       " 'CCMSLIB00005436031',\n",
       " 'CCMSLIB00005436032',\n",
       " 'CCMSLIB00005436033',\n",
       " 'CCMSLIB00005436034',\n",
       " 'CCMSLIB00005436071',\n",
       " 'CCMSLIB00005436072',\n",
       " 'CCMSLIB00005436074',\n",
       " 'CCMSLIB00005436075',\n",
       " 'CCMSLIB00005436076',\n",
       " 'CCMSLIB00005436077',\n",
       " 'CCMSLIB00005436085',\n",
       " 'CCMSLIB00005436086',\n",
       " 'CCMSLIB00005436459',\n",
       " 'CCMSLIB00005436460',\n",
       " 'CCMSLIB00005436463',\n",
       " 'CCMSLIB00005436466',\n",
       " 'CCMSLIB00005436470',\n",
       " 'CCMSLIB00005436494',\n",
       " 'CCMSLIB00005436496',\n",
       " 'CCMSLIB00005436497',\n",
       " 'CCMSLIB00005463461',\n",
       " 'CCMSLIB00005463462',\n",
       " 'CCMSLIB00005463464',\n",
       " 'CCMSLIB00005463465',\n",
       " 'CCMSLIB00005463466',\n",
       " 'CCMSLIB00005463467',\n",
       " 'CCMSLIB00005463468',\n",
       " 'CCMSLIB00005463469',\n",
       " 'CCMSLIB00005463470',\n",
       " 'CCMSLIB00005463471',\n",
       " 'CCMSLIB00005463472',\n",
       " 'CCMSLIB00005463473',\n",
       " 'CCMSLIB00005463474',\n",
       " 'CCMSLIB00005463475',\n",
       " 'CCMSLIB00005463477',\n",
       " 'CCMSLIB00005463853',\n",
       " 'CCMSLIB00005463854',\n",
       " 'CCMSLIB00005463855',\n",
       " 'CCMSLIB00005463857',\n",
       " 'CCMSLIB00005463858',\n",
       " 'CCMSLIB00005463859',\n",
       " 'CCMSLIB00005464636',\n",
       " 'CCMSLIB00005464637',\n",
       " 'CCMSLIB00005464638',\n",
       " 'CCMSLIB00005464639',\n",
       " 'CCMSLIB00005464640',\n",
       " 'CCMSLIB00005464641',\n",
       " 'CCMSLIB00005464642',\n",
       " 'CCMSLIB00005464643',\n",
       " 'CCMSLIB00005466096',\n",
       " 'CCMSLIB00005466097',\n",
       " 'CCMSLIB00005467777',\n",
       " 'CCMSLIB00005467778',\n",
       " 'CCMSLIB00005467779',\n",
       " 'CCMSLIB00005467780',\n",
       " 'CCMSLIB00005467781',\n",
       " 'CCMSLIB00005467782',\n",
       " 'CCMSLIB00005467783',\n",
       " 'CCMSLIB00005467784',\n",
       " 'CCMSLIB00005467785',\n",
       " 'CCMSLIB00005467786',\n",
       " 'CCMSLIB00005467787',\n",
       " 'CCMSLIB00005467788',\n",
       " 'CCMSLIB00005467789',\n",
       " 'CCMSLIB00005467790',\n",
       " 'CCMSLIB00005467791',\n",
       " 'CCMSLIB00005467792',\n",
       " 'CCMSLIB00005467793',\n",
       " 'CCMSLIB00005467794',\n",
       " 'CCMSLIB00005467795',\n",
       " 'CCMSLIB00005467796',\n",
       " 'CCMSLIB00005467800',\n",
       " 'CCMSLIB00005467801',\n",
       " 'CCMSLIB00005467802',\n",
       " 'CCMSLIB00005467803',\n",
       " 'CCMSLIB00005720212',\n",
       " 'CCMSLIB00005720213',\n",
       " 'CCMSLIB00005720214',\n",
       " 'CCMSLIB00005720215',\n",
       " 'CCMSLIB00005720216',\n",
       " 'CCMSLIB00005720217',\n",
       " 'CCMSLIB00005720230',\n",
       " 'CCMSLIB00005720231',\n",
       " 'CCMSLIB00005720233',\n",
       " 'CCMSLIB00005720234',\n",
       " 'CCMSLIB00005720235',\n",
       " 'CCMSLIB00005720236',\n",
       " 'CCMSLIB00005721064',\n",
       " 'CCMSLIB00005721065',\n",
       " 'CCMSLIB00005723386',\n",
       " 'CCMSLIB00005723387',\n",
       " 'CCMSLIB00005723388',\n",
       " 'CCMSLIB00005723389',\n",
       " 'CCMSLIB00005723390',\n",
       " 'CCMSLIB00005723391',\n",
       " 'CCMSLIB00005723392',\n",
       " 'CCMSLIB00005723393',\n",
       " 'CCMSLIB00005723394',\n",
       " 'CCMSLIB00005723395',\n",
       " 'CCMSLIB00005723396',\n",
       " 'CCMSLIB00005723397',\n",
       " 'CCMSLIB00005723398',\n",
       " 'CCMSLIB00005723399',\n",
       " 'CCMSLIB00005723400',\n",
       " 'CCMSLIB00005723401',\n",
       " 'CCMSLIB00005723402',\n",
       " 'CCMSLIB00005723403',\n",
       " 'CCMSLIB00005723404',\n",
       " 'CCMSLIB00005723405',\n",
       " 'CCMSLIB00005723406',\n",
       " 'CCMSLIB00005723407',\n",
       " 'CCMSLIB00005723408',\n",
       " 'CCMSLIB00005723409',\n",
       " 'CCMSLIB00005723410',\n",
       " 'CCMSLIB00005723411',\n",
       " 'CCMSLIB00005723412',\n",
       " 'CCMSLIB00005723413',\n",
       " 'CCMSLIB00005723414',\n",
       " 'CCMSLIB00005723415',\n",
       " 'CCMSLIB00005723416',\n",
       " 'CCMSLIB00005723417',\n",
       " 'CCMSLIB00005723418',\n",
       " 'CCMSLIB00005723419',\n",
       " 'CCMSLIB00005723420',\n",
       " 'CCMSLIB00005723986',\n",
       " 'CCMSLIB00005724004',\n",
       " 'CCMSLIB00005724017',\n",
       " 'CCMSLIB00005724062']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "select_list = []\n",
    "\n",
    "for item in gnps_df2['SPECTRUMID']:\n",
    "    select_list.append(item)\n",
    "\n",
    "select_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "with mgf.MGF('/Users/tiagoferreiraleao/Dropbox/tiago-NAS/NPOmix/inputs/ALL_GNPS.mgf') as reader:\n",
    "    for spectrum in reader:\n",
    "        if spectrum['params']['spectrumid'] in select_list:\n",
    "            individual_spec = []\n",
    "            individual_spec.append(spectrum)\n",
    "            output_file = '/Users/tiagoferreiraleao/Dropbox/tiago-NAS/NPOmix/NPOmix_mgf_spectra/%s.mgf'%(spectrum['params']['spectrumid'])\n",
    "#             mgf.write(spectra=individual_spec, header='', output=output_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     406\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Users/tiagoferreiraleao/Dropbox/tiago-NAS/NPOmix/NPOmix_mgf_spectra/ | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome/Metagenome</th>\n",
       "      <th>Location of metabolomics data file</th>\n",
       "      <th>NCBI ID</th>\n",
       "      <th>Sample Growth Conditions</th>\n",
       "      <th>Extraction Method</th>\n",
       "      <th>Instrumentation Method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Symploca sp. SIO1B1</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010672945</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Cyanothece sp. SIO1E1</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010672835</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Moorea sp. SIO1F2</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010672755</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Okeania sp. SIO1F9</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010672745</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Okeania sp. SIO1F9</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010672745</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>Leptolyngbya sp. SIO4C5</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010671965</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>Moorea sp. SIO4G2</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010692345</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Moorea sp. SIO4G3</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010692305</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Moorea sp. SIOASIH</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010671925</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>Leptolyngbya sp. SIOISBB</td>\n",
       "      <td>ftp://massive.ucsd.edu/MSV000085210/raw/iomega...</td>\n",
       "      <td>GCA_010671905</td>\n",
       "      <td>SW-BG-11</td>\n",
       "      <td>2:1 DCM:MeOH</td>\n",
       "      <td>Maxis LCMS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Genome/Metagenome  \\\n",
       "0         Symploca sp. SIO1B1   \n",
       "1       Cyanothece sp. SIO1E1   \n",
       "2           Moorea sp. SIO1F2   \n",
       "3          Okeania sp. SIO1F9   \n",
       "4          Okeania sp. SIO1F9   \n",
       "..                        ...   \n",
       "109   Leptolyngbya sp. SIO4C5   \n",
       "110         Moorea sp. SIO4G2   \n",
       "111         Moorea sp. SIO4G3   \n",
       "112        Moorea sp. SIOASIH   \n",
       "113  Leptolyngbya sp. SIOISBB   \n",
       "\n",
       "                    Location of metabolomics data file        NCBI ID  \\\n",
       "0    ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010672945   \n",
       "1    ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010672835   \n",
       "2    ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010672755   \n",
       "3    ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010672745   \n",
       "4    ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010672745   \n",
       "..                                                 ...            ...   \n",
       "109  ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010671965   \n",
       "110  ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010692345   \n",
       "111  ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010692305   \n",
       "112  ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010671925   \n",
       "113  ftp://massive.ucsd.edu/MSV000085210/raw/iomega...  GCA_010671905   \n",
       "\n",
       "    Sample Growth Conditions Extraction Method Instrumentation Method  \n",
       "0                   SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "1                   SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "2                   SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "3                   SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "4                   SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "..                       ...               ...                    ...  \n",
       "109                 SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "110                 SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "111                 SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "112                 SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "113                 SW-BG-11      2:1 DCM:MeOH             Maxis LCMS  \n",
       "\n",
       "[114 rows x 6 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#downloaded file from the Gerwick dataset with \n",
    "podp_lena_df = pd.read_csv('/Users/tiagoferreiraleao/Dropbox/tiago-NAS/NPOmix/inputs/paired-864909ec-e716-4c5a-bfe3-ce3a169b8844.2-genome-metabolome.csv',sep=',')\n",
    "\n",
    "podp_lena_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "114"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows_wo_file,seen,commands = [],[],[]\n",
    "\n",
    "for i,r in podp_lena_df.iterrows():\n",
    "    genomeID = podp_lena_df['NCBI ID'].loc[i]\n",
    "    seen.append(genomeID)\n",
    "    file_count = seen.count(genomeID)\n",
    "    extension = podp_lena_df['Location of metabolomics data file'].loc[i].rsplit('.', 1)[1]\n",
    "    cmd = 'wget -O %s %s'%(genomeID+'.'+extension+'.'+str(file_count),podp_lena_df['Location of metabolomics data file'].loc[i])\n",
    "    commands.append(cmd)\n",
    "\n",
    "len(commands)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['wget -O GCA_010672945.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2248%20cr_GC6_01_37750.mzXML',\n",
       " 'wget -O GCA_010672835.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/PAB18MAY11_9_C_GA10_01_37730.mzXML',\n",
       " 'wget -O GCA_010672755.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2239%20cr_GG1_01_37793.mzXML',\n",
       " 'wget -O GCA_010672745.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2058%20D_GD8_01_37764.mzXML',\n",
       " 'wget -O GCA_010672745.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2058%20H_GD9_01_37765.mzXML',\n",
       " 'wget -O GCA_010672795.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2035%20cr_GD1_01_37757.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015E11_RE11_01_13970.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1766_015F11_RF11_01_13982.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F2_RF2_01_13972.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F3_RF3_01_13973.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F4_RF4_01_13974.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F5_RF5_01_13975.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F6_RF6_01_13976.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F7_RF7_01_13977.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.9 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F8_RF8_01_13978.mzXML',\n",
       " 'wget -O GCA_010672695.mzXML.10 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1765_015F9_RF9_01_13979.mzXML',\n",
       " 'wget -O GCA_010672535.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2046%20cr_GD7_01_37763.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F2_RF2_01_13801.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F3_RF3_01_13805.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F4_RF4_01_13806.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F5_RF5_01_13807.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F6_RF6_01_13808.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F7_RF7_01_13809.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F8_RF8_01_13810.mzXML',\n",
       " 'wget -O GCA_010672505.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1736_013F10_RF10_01_13812.mzXML',\n",
       " 'wget -O GCA_010692635.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2080%20cr_GE2_01_37770.mzXML',\n",
       " 'wget -O GCA_010672485.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2093%20cr_GE5_01_37773.mzXML',\n",
       " 'wget -O GCA_010672455.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1952_028A5_GA5_01_15275.mzXML',\n",
       " 'wget -O GCA_010672455.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1952_028A6_GA6_01_15276.mzXML',\n",
       " 'wget -O GCA_010672455.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1952_028A7_2_GA7_01_15279.mzXML',\n",
       " 'wget -O GCA_010672455.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1952_028A8_GA8_01_15278.mzXML',\n",
       " 'wget -O GCA_010672455.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1952_028A9_GA9_01_15281.mzXML',\n",
       " 'wget -O GCA_010672455.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1952_028A10_GA10_01_15282.mzXML',\n",
       " 'wget -O GCA_010672455.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1952_028A11_GA11_01_15283.mzXML',\n",
       " 'wget -O GCA_010672405.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1831_018A6_GA6_01_14226.mzXML',\n",
       " 'wget -O GCA_010692645.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1716_012E3_GE3_01_13654.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027E8_RE8_01_15222.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027E9_RE9_01_15223.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027E10_RE10_01_15224.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027E11_RE11_01_15225.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1944_027F10_RF10_01_15236.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027F2_RF2_01_15227.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027F3_RF3_01_15228.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027F4_RF4_01_15229.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.9 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027F5_RF5_01_15230.mzXML',\n",
       " 'wget -O GCA_010672385.mzXML.10 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1943_027F6_RF6_01_15231.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027D9_RD9_01_15211.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027D10_RD10_01_15212.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027D11_RD11_01_15213.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027E2_RE2_01_15215.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027E3_RE3_01_15216.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027E4_RE4_01_15217.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027E5_RE5_01_15218.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027E6_RE6_01_15219.mzXML',\n",
       " 'wget -O GCA_010692685.mzXML.9 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1942_027E7_RE7_01_15220.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024A5_GA5_01_14839.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024A6_GA6_01_14840.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024A7_GA7_01_14841.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024A8_GA8_01_14842.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024A9_GA9_01_14843.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024A10_GA10_01_14844.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024A11_GA11_01_14845.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/DMSO_1893_024B12_GB12_01_14852.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.9 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024B2_GB2_01_14847.mzXML',\n",
       " 'wget -O GCA_010672365.mzXML.10 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1891_024B3_GB3_01_14848.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F2_GF2_01_14710.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F3_GF3_01_14711.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F4_GF4_01_14712.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F5_GF5_01_14713.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F6_GF6_01_14714.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F7_GF7_01_14715.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F9_GF9_01_14717.mzXML',\n",
       " 'wget -O GCA_010672305.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1872_022F10_GF10_01_14718.mzXML',\n",
       " 'wget -O GCA_010672345.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1889_024A3_GA3_01_14835.mzXML',\n",
       " 'wget -O GCA_010692575.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1894_024C4_GC4_01_14864.mzXML',\n",
       " 'wget -O GCA_010692585.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/PAL24MAY13_5_C_GB5_01_37737.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1902_024H11_GH11_01_14948.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H2_GH2_01_14937.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H3_GH3_01_14938.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H4_GH4_01_14939.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H5_GH5_01_14940.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H6_GH6_01_14941.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H7_GH7_01_14942.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H8_GH8_01_14944.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.9 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H9_GH9_01_14945.mzXML',\n",
       " 'wget -O GCA_010672205.mzXML.10 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/1901_024H10_GH10_01_14946.mzXML',\n",
       " 'wget -O GCA_010692525.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1849_020H9_GH9_01_14532.mzXML',\n",
       " 'wget -O GCA_010692535.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2098%20cr_GE6_01_37774.mzXML',\n",
       " 'wget -O GCA_010692555.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/DMSO_1752_014B12_GB12_01_13869.mzXML',\n",
       " 'wget -O GCA_010692465.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2070%20E_GD11_01_37767.mzXML',\n",
       " 'wget -O GCA_010672215.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2079%20cr_GE1_01_37769.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1837_020A11_GA11_01_14445.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A2_GA2_01_14435.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A3_GA3_01_14436.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A4_GA4_01_14437.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A5_GA5_01_14438.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.6 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A6_GA6_01_14439.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.7 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A7_GA7_01_14440.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.8 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A8_GA8_01_14441.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.9 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A9_GA9_01_14442.mzXML',\n",
       " 'wget -O GCA_010692425.mzXML.10 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1833_020A10_GA10_01_14443.mzXML',\n",
       " 'wget -O GCA_010672195.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1688_010E7_GE7_01_13470.mzXML',\n",
       " 'wget -O GCA_010672195.mzXML.2 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1688_010E8_GE8_01_13471.mzXML',\n",
       " 'wget -O GCA_010672195.mzXML.3 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1688_010E9_GE9_01_13472.mzXML',\n",
       " 'wget -O GCA_010672195.mzXML.4 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1688_010E10_GE10_01_13473.mzXML',\n",
       " 'wget -O GCA_010672195.mzXML.5 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/A1688_010E11_GE11_01_13474.mzXML',\n",
       " 'wget -O GCA_010692445.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2103%20cr_GE7_01_37775.mzXML',\n",
       " 'wget -O GCA_010671975.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/Cobble%20down%20cr_GG6_01_37798.mzXML',\n",
       " 'wget -O GCA_010692335.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/numC_11_C_GC2_01_37746.mzXML',\n",
       " 'wget -O GCA_010671965.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/C15-B%20cr_GG7_01_37799.mzXML',\n",
       " 'wget -O GCA_010692345.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2246%20cr_GG3_01_37795.mzXML',\n",
       " 'wget -O GCA_010692305.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/2247%20cr_GG4_01_37796.mzXML',\n",
       " 'wget -O GCA_010671925.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/ASI16JUL14_2_C_GA3_01_37723.mzXML',\n",
       " 'wget -O GCA_010671905.mzXML.1 ftp://massive.ucsd.edu/MSV000085210/raw/iomega/ISB03NOV94_8_C_GA6_01_37726.mzXML']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "commands"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "### only run once\n",
    "\n",
    "# with open('./temp_files/PODP_LCMS_list_commands-round5.txt', \"w\") as output:\n",
    "#     writer = csv.writer(output, lineterminator='\\n')\n",
    "#     for val in commands:\n",
    "#         writer.writerow([val])\n",
    "\n",
    "# !mkdir ./inputs/cyanos_mzML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     118\r\n"
     ]
    }
   ],
   "source": [
    "!ls ./inputs/cyanos_mzML/ | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!rm ./temp_files/PODP_LCMS_list_commands-round5.txt"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
